// @searchapi/n8n-nodes-searchapi/dist/credentials/SearchApi.credentials.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchApi = void 0;
class SearchApi {
    constructor() {
        this.name = 'searchApi';
        this.displayName = 'SearchApi API';
        this.documentationUrl = 'https://www.searchapi.io/docs/google';
        this.icon = 'file:../nodes/SearchApi/searchApi.svg';
        this.properties = [
            {
                displayName: 'API Key',
                name: 'apiKey',
                type: 'string',
                typeOptions: { password: true },
                default: '',
                required: true,
            },
        ];
        this.authenticate = {
            type: 'generic',
            properties: {
                qs: { api_key: '={{ $credentials.apiKey }}' },
                headers: {
                    "X-SearchApi-Source": "N8N"
                }
            }
        };
        this.test = {
            request: {
                baseURL: 'https://www.searchapi.io/api/v1/me',
                method: 'GET',
                qs: { api_key: '={{ $credentials.apiKey }}' },
                headers: {
                    "X-SearchApi-Source": "N8N"
                }
            }
        };
    }
}
exports.SearchApi = SearchApi;
