// @searchapi/n8n-nodes-searchapi/dist/nodes/SearchApi/SearchApi.node.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchApi = void 0;
const google_1 = require("./engines/google");
const google_images_1 = require("./engines/google_images");
const google_maps_1 = require("./engines/google_maps");
const google_shopping_1 = require("./engines/google_shopping");
class SearchApi {
    constructor() {
        this.description = {
            displayName: 'SearchApi',
            name: 'searchApi',
            icon: 'file:searchApi.svg',
            group: ['output'],
            version: 1,
            description: 'Access real-time search results from Google, Google Images, Google Maps, Google Shopping and more. Use this when you need current, up-to-date information, product searches, location data, or visual content that may not be available in your training data.',
            subtitle: '={{ $parameter["engine"] }}',
            defaults: { name: 'SearchApi' },
            inputs: ["main"],
            outputs: ["main"],
            credentials: [{ name: 'searchApi', required: true }],
            usableAsTool: true,
            requestDefaults: {
                baseURL: 'https://www.searchapi.io/api/v1',
                method: 'GET',
                url: '/search',
                headers: { Accept: 'application/json' },
            },
            hints: [
                {
                    message: "Hit SearchAPI's free 100-request quota? Check the Pricing page 📈'",
                    type: 'info',
                    whenToDisplay: 'beforeExecution',
                    location: 'inputPane',
                },
            ],
            properties: [
                {
                    displayName: 'Resource',
                    name: 'resource',
                    type: 'options',
                    description: 'The search engine to use',
                    noDataExpression: true,
                    options: [
                        google_1.google.resource,
                        google_images_1.google_images.resource,
                        google_maps_1.google_maps.resource,
                        google_shopping_1.google_shopping.resource,
                    ],
                    default: google_1.google.resource.value,
                },
                {
                    displayName: 'Operation Name',
                    name: 'operation',
                    type: 'options',
                    noDataExpression: true,
                    options: [
                        {
                            name: 'Search',
                            value: 'search',
                            action: 'Search',
                            description: 'Search using the engine specified in the resource',
                            routing: {
                                request: {
                                    qs: {
                                        engine: '={{ $parameter["resource"] }}',
                                    },
                                },
                            },
                        },
                    ],
                    default: 'search',
                },
                ...google_1.google.properties,
                ...google_images_1.google_images.properties,
                ...google_maps_1.google_maps.properties,
                ...google_shopping_1.google_shopping.properties,
            ],
        };
    }
}
exports.SearchApi = SearchApi;
