// @searchapi/n8n-nodes-searchapi/dist/nodes/SearchApi/engines/google.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.google = void 0;
const displayOptions = {
    show: {
        resource: ['google'],
    },
};
const resource = {
    name: 'Google Search',
    value: 'google'
};
const properties = [
    {
        displayName: 'Query',
        name: 'q',
        type: 'string',
        default: '',
        required: true,
        description: 'Parameter defines the query you want to search. You can use anything that you would use in a regular Google search. e.g. inurl:, site:, intitle:, and more.',
        options: [],
        displayOptions,
        routing: {
            request: {
                qs: {
                    q: '={{$value}}',
                },
            },
        },
    },
    {
        displayName: 'Location Settings',
        name: 'locationSettings',
        type: 'collection',
        placeholder: 'Add Location Setting',
        default: {},
        options: [
            {
                displayName: 'Country',
                name: 'gl',
                type: 'options',
                default: 'us',
                description: 'The default parameter us defines the country of the search. Check the full list of supported Google gl countries.',
                options: [
                    { name: 'Afghanistan', value: 'af' },
                    { name: 'Albania', value: 'al' },
                    { name: 'Algeria', value: 'dz' },
                    { name: 'American Samoa', value: 'as' },
                    { name: 'Andorra', value: 'ad' },
                    { name: 'Angola', value: 'ao' },
                    { name: 'Anguilla', value: 'ai' },
                    { name: 'Antarctica', value: 'aq' },
                    { name: 'Antigua and Barbuda', value: 'ag' },
                    { name: 'Argentina', value: 'ar' },
                    { name: 'Armenia', value: 'am' },
                    { name: 'Aruba', value: 'aw' },
                    { name: 'Australia', value: 'au' },
                    { name: 'Austria', value: 'at' },
                    { name: 'Azerbaijan', value: 'az' },
                    { name: 'Bahamas', value: 'bs' },
                    { name: 'Bahrain', value: 'bh' },
                    { name: 'Bangladesh', value: 'bd' },
                    { name: 'Barbados', value: 'bb' },
                    { name: 'Belarus', value: 'by' },
                    { name: 'Belgium', value: 'be' },
                    { name: 'Belize', value: 'bz' },
                    { name: 'Benin', value: 'bj' },
                    { name: 'Bermuda', value: 'bm' },
                    { name: 'Bhutan', value: 'bt' },
                    { name: 'Bolivia', value: 'bo' },
                    { name: 'Bosnia and Herzegovina', value: 'ba' },
                    { name: 'Botswana', value: 'bw' },
                    { name: 'Bouvet Island', value: 'bv' },
                    { name: 'Brazil', value: 'br' },
                    { name: 'British Indian Ocean Territory', value: 'io' },
                    { name: 'British Virgin Islands', value: 'vg' },
                    { name: 'Brunei', value: 'bn' },
                    { name: 'Bulgaria', value: 'bg' },
                    { name: 'Burkina Faso', value: 'bf' },
                    { name: 'Burundi', value: 'bi' },
                    { name: 'Cambodia', value: 'kh' },
                    { name: 'Cameroon', value: 'cm' },
                    { name: 'Canada', value: 'ca' },
                    { name: 'Cape Verde', value: 'cv' },
                    { name: 'Cayman Islands', value: 'ky' },
                    { name: 'Central African Republic', value: 'cf' },
                    { name: 'Chad', value: 'td' },
                    { name: 'Chile', value: 'cl' },
                    { name: 'China', value: 'cn' },
                    { name: 'Christmas Island', value: 'cx' },
                    { name: 'Cocos Islands', value: 'cc' },
                    { name: 'Colombia', value: 'co' },
                    { name: 'Comoros', value: 'km' },
                    { name: 'Cook Islands', value: 'ck' },
                    { name: 'Costa Rica', value: 'cr' },
                    { name: 'Côte d\'Ivoire', value: 'ci' },
                    { name: 'Croatia', value: 'hr' },
                    { name: 'Cuba', value: 'cu' },
                    { name: 'Cyprus', value: 'cy' },
                    { name: 'Czech Republic', value: 'cz' },
                    { name: 'Democratic Republic of the Congo', value: 'cd' },
                    { name: 'Denmark', value: 'dk' },
                    { name: 'Djibouti', value: 'dj' },
                    { name: 'Dominica', value: 'dm' },
                    { name: 'Dominican Republic', value: 'do' },
                    { name: 'Ecuador', value: 'ec' },
                    { name: 'Egypt', value: 'eg' },
                    { name: 'El Salvador', value: 'sv' },
                    { name: 'Equatorial Guinea', value: 'gq' },
                    { name: 'Eritrea', value: 'er' },
                    { name: 'Estonia', value: 'ee' },
                    { name: 'Eswatini', value: 'sz' },
                    { name: 'Ethiopia', value: 'et' },
                    { name: 'Falkland Islands', value: 'fk' },
                    { name: 'Faroe Islands', value: 'fo' },
                    { name: 'Fiji', value: 'fj' },
                    { name: 'Finland', value: 'fi' },
                    { name: 'France', value: 'fr' },
                    { name: 'French Guiana', value: 'gf' },
                    { name: 'French Polynesia', value: 'pf' },
                    { name: 'French Southern Territories', value: 'tf' },
                    { name: 'Gabon', value: 'ga' },
                    { name: 'Gambia', value: 'gm' },
                    { name: 'Georgia', value: 'ge' },
                    { name: 'Germany', value: 'de' },
                    { name: 'Ghana', value: 'gh' },
                    { name: 'Gibraltar', value: 'gi' },
                    { name: 'Greece', value: 'gr' },
                    { name: 'Greenland', value: 'gl' },
                    { name: 'Grenada', value: 'gd' },
                    { name: 'Guadeloupe', value: 'gp' },
                    { name: 'Guam', value: 'gu' },
                    { name: 'Guatemala', value: 'gt' },
                    { name: 'Guernsey', value: 'gg' },
                    { name: 'Guinea', value: 'gn' },
                    { name: 'Guinea-Bissau', value: 'gw' },
                    { name: 'Guyana', value: 'gy' },
                    { name: 'Haiti', value: 'ht' },
                    { name: 'Heard Island and McDonald Islands', value: 'hm' },
                    { name: 'Honduras', value: 'hn' },
                    { name: 'Hong Kong', value: 'hk' },
                    { name: 'Hungary', value: 'hu' },
                    { name: 'Iceland', value: 'is' },
                    { name: 'India', value: 'in' },
                    { name: 'Indonesia', value: 'id' },
                    { name: 'Iran', value: 'ir' },
                    { name: 'Iraq', value: 'iq' },
                    { name: 'Ireland', value: 'ie' },
                    { name: 'Isle of Man', value: 'im' },
                    { name: 'Israel', value: 'il' },
                    { name: 'Italy', value: 'it' },
                    { name: 'Jamaica', value: 'jm' },
                    { name: 'Japan', value: 'jp' },
                    { name: 'Jersey', value: 'je' },
                    { name: 'Jordan', value: 'jo' },
                    { name: 'Kazakhstan', value: 'kz' },
                    { name: 'Kenya', value: 'ke' },
                    { name: 'Kiribati', value: 'ki' },
                    { name: 'Kuwait', value: 'kw' },
                    { name: 'Kyrgyzstan', value: 'kg' },
                    { name: 'Laos', value: 'la' },
                    { name: 'Latvia', value: 'lv' },
                    { name: 'Lebanon', value: 'lb' },
                    { name: 'Lesotho', value: 'ls' },
                    { name: 'Liberia', value: 'lr' },
                    { name: 'Libya', value: 'ly' },
                    { name: 'Liechtenstein', value: 'li' },
                    { name: 'Lithuania', value: 'lt' },
                    { name: 'Luxembourg', value: 'lu' },
                    { name: 'Macao', value: 'mo' },
                    { name: 'Madagascar', value: 'mg' },
                    { name: 'Malawi', value: 'mw' },
                    { name: 'Malaysia', value: 'my' },
                    { name: 'Maldives', value: 'mv' },
                    { name: 'Mali', value: 'ml' },
                    { name: 'Malta', value: 'mt' },
                    { name: 'Marshall Islands', value: 'mh' },
                    { name: 'Martinique', value: 'mq' },
                    { name: 'Mauritania', value: 'mr' },
                    { name: 'Mauritius', value: 'mu' },
                    { name: 'Mayotte', value: 'yt' },
                    { name: 'Mexico', value: 'mx' },
                    { name: 'Micronesia', value: 'fm' },
                    { name: 'Moldova', value: 'md' },
                    { name: 'Monaco', value: 'mc' },
                    { name: 'Mongolia', value: 'mn' },
                    { name: 'Montenegro', value: 'me' },
                    { name: 'Montserrat', value: 'ms' },
                    { name: 'Morocco', value: 'ma' },
                    { name: 'Mozambique', value: 'mz' },
                    { name: 'Myanmar', value: 'mm' },
                    { name: 'Namibia', value: 'na' },
                    { name: 'Nauru', value: 'nr' },
                    { name: 'Nepal', value: 'np' },
                    { name: 'Netherlands', value: 'nl' },
                    { name: 'New Caledonia', value: 'nc' },
                    { name: 'New Zealand', value: 'nz' },
                    { name: 'Nicaragua', value: 'ni' },
                    { name: 'Niger', value: 'ne' },
                    { name: 'Nigeria', value: 'ng' },
                    { name: 'Niue', value: 'nu' },
                    { name: 'Norfolk Island', value: 'nf' },
                    { name: 'North Korea', value: 'kp' },
                    { name: 'North Macedonia', value: 'mk' },
                    { name: 'Northern Mariana Islands', value: 'mp' },
                    { name: 'Norway', value: 'no' },
                    { name: 'Oman', value: 'om' },
                    { name: 'Pakistan', value: 'pk' },
                    { name: 'Palau', value: 'pw' },
                    { name: 'Palestine', value: 'ps' },
                    { name: 'Panama', value: 'pa' },
                    { name: 'Papua New Guinea', value: 'pg' },
                    { name: 'Paraguay', value: 'py' },
                    { name: 'Peru', value: 'pe' },
                    { name: 'Philippines', value: 'ph' },
                    { name: 'Pitcairn', value: 'pn' },
                    { name: 'Poland', value: 'pl' },
                    { name: 'Portugal', value: 'pt' },
                    { name: 'Puerto Rico', value: 'pr' },
                    { name: 'Qatar', value: 'qa' },
                    { name: 'Republic of the Congo', value: 'cg' },
                    { name: 'Réunion', value: 're' },
                    { name: 'Romania', value: 'ro' },
                    { name: 'Russia', value: 'ru' },
                    { name: 'Rwanda', value: 'rw' },
                    { name: 'Saint Helena', value: 'sh' },
                    { name: 'Saint Kitts and Nevis', value: 'kn' },
                    { name: 'Saint Lucia', value: 'lc' },
                    { name: 'Saint Pierre and Miquelon', value: 'pm' },
                    { name: 'Saint Vincent and the Grenadines', value: 'vc' },
                    { name: 'Samoa', value: 'ws' },
                    { name: 'San Marino', value: 'sm' },
                    { name: 'São Tomé and Príncipe', value: 'st' },
                    { name: 'Saudi Arabia', value: 'sa' },
                    { name: 'Senegal', value: 'sn' },
                    { name: 'Serbia', value: 'rs' },
                    { name: 'Seychelles', value: 'sc' },
                    { name: 'Sierra Leone', value: 'sl' },
                    { name: 'Singapore', value: 'sg' },
                    { name: 'Slovakia', value: 'sk' },
                    { name: 'Slovenia', value: 'si' },
                    { name: 'Solomon Islands', value: 'sb' },
                    { name: 'Somalia', value: 'so' },
                    { name: 'South Africa', value: 'za' },
                    { name: 'South Georgia and the South Sandwich Islands', value: 'gs' },
                    { name: 'South Korea', value: 'kr' },
                    { name: 'Spain', value: 'es' },
                    { name: 'Sri Lanka', value: 'lk' },
                    { name: 'Sudan', value: 'sd' },
                    { name: 'Suriname', value: 'sr' },
                    { name: 'Svalbard and Jan Mayen', value: 'sj' },
                    { name: 'Sweden', value: 'se' },
                    { name: 'Switzerland', value: 'ch' },
                    { name: 'Taiwan', value: 'tw' },
                    { name: 'Tajikistan', value: 'tj' },
                    { name: 'Tanzania', value: 'tz' },
                    { name: 'Thailand', value: 'th' },
                    { name: 'Timor-Leste', value: 'tl' },
                    { name: 'Togo', value: 'tg' },
                    { name: 'Tokelau', value: 'tk' },
                    { name: 'Tonga', value: 'to' },
                    { name: 'Trinidad and Tobago', value: 'tt' },
                    { name: 'Tunisia', value: 'tn' },
                    { name: 'Turkey', value: 'tr' },
                    { name: 'Turkmenistan', value: 'tm' },
                    { name: 'Turks and Caicos Islands', value: 'tc' },
                    { name: 'Tuvalu', value: 'tv' },
                    { name: 'U.S. Virgin Islands', value: 'vi' },
                    { name: 'Uganda', value: 'ug' },
                    { name: 'Ukraine', value: 'ua' },
                    { name: 'United Arab Emirates', value: 'ae' },
                    { name: 'United Kingdom', value: 'uk' },
                    { name: 'United States', value: 'us' },
                    { name: 'United States Minor Outlying Islands', value: 'um' },
                    { name: 'Uruguay', value: 'uy' },
                    { name: 'Uzbekistan', value: 'uz' },
                    { name: 'Vanuatu', value: 'vu' },
                    { name: 'Vatican City', value: 'va' },
                    { name: 'Venezuela', value: 've' },
                    { name: 'Vietnam', value: 'vn' },
                    { name: 'Wallis and Futuna', value: 'wf' },
                    { name: 'Western Sahara', value: 'eh' },
                    { name: 'Yemen', value: 'ye' },
                    { name: 'Zambia', value: 'zm' },
                    { name: 'Zimbabwe', value: 'zw' },
                ],
                routing: {
                    request: {
                        qs: {
                            gl: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Country Restriction',
                name: 'cr',
                type: 'string',
                default: '',
                description: 'The cr parameter restricts search results to documents originating in a particular country. Google determines the country of a document by the top-level domain (TLD) of the document\'s URL or by Web server\'s IP address geographic location. Check the full list of supported Google cr countries.',
                routing: {
                    request: {
                        qs: {
                            cr: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Encoded Location',
                name: 'uule',
                type: 'string',
                default: '',
                description: 'Parameter is the Google encoded location you want to use for the search. SearchApi automatically generated the uule parameter when you use the location parameter but we allow you to overwrite it directly. uule and location parameters can\'t be used together.',
                routing: {
                    request: {
                        qs: {
                            uule: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Geographic Location',
                name: 'location',
                type: 'string',
                default: '',
                description: 'Parameter defines from where you want the search to originate. If several locations match the location requested, we\'ll pick the most popular one. Head to the Locations API if you need more precise control.',
                routing: {
                    request: {
                        qs: {
                            location: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Google Domain',
                name: 'google_domain',
                type: 'options',
                default: 'google.com',
                description: 'The default parameter google.com defines the Google domain of the search. Check the full list of supported Google google_domain domains.',
                options: [
                    { name: 'Google.ad', value: 'google.ad' },
                    { name: 'Google.ae', value: 'google.ae' },
                    { name: 'Google.al', value: 'google.al' },
                    { name: 'Google.am', value: 'google.am' },
                    { name: 'Google.as', value: 'google.as' },
                    { name: 'Google.at', value: 'google.at' },
                    { name: 'Google.az', value: 'google.az' },
                    { name: 'Google.ba', value: 'google.ba' },
                    { name: 'Google.be', value: 'google.be' },
                    { name: 'Google.bf', value: 'google.bf' },
                    { name: 'Google.bg', value: 'google.bg' },
                    { name: 'Google.bi', value: 'google.bi' },
                    { name: 'Google.bj', value: 'google.bj' },
                    { name: 'Google.bs', value: 'google.bs' },
                    { name: 'Google.bt', value: 'google.bt' },
                    { name: 'Google.by', value: 'google.by' },
                    { name: 'Google.ca', value: 'google.ca' },
                    { name: 'Google.cd', value: 'google.cd' },
                    { name: 'Google.cf', value: 'google.cf' },
                    { name: 'Google.cg', value: 'google.cg' },
                    { name: 'Google.ch', value: 'google.ch' },
                    { name: 'Google.ci', value: 'google.ci' },
                    { name: 'Google.cl', value: 'google.cl' },
                    { name: 'Google.cm', value: 'google.cm' },
                    { name: 'Google.co.ao', value: 'google.co.ao' },
                    { name: 'Google.co.bw', value: 'google.co.bw' },
                    { name: 'Google.co.ck', value: 'google.co.ck' },
                    { name: 'Google.co.cr', value: 'google.co.cr' },
                    { name: 'Google.co.ID', value: 'google.co.id' },
                    { name: 'Google.co.il', value: 'google.co.il' },
                    { name: 'Google.co.in', value: 'google.co.in' },
                    { name: 'Google.co.jp', value: 'google.co.jp' },
                    { name: 'Google.co.ke', value: 'google.co.ke' },
                    { name: 'Google.co.kr', value: 'google.co.kr' },
                    { name: 'Google.co.ls', value: 'google.co.ls' },
                    { name: 'Google.co.ma', value: 'google.co.ma' },
                    { name: 'Google.co.mz', value: 'google.co.mz' },
                    { name: 'Google.co.nz', value: 'google.co.nz' },
                    { name: 'Google.co.th', value: 'google.co.th' },
                    { name: 'Google.co.tz', value: 'google.co.tz' },
                    { name: 'Google.co.ug', value: 'google.co.ug' },
                    { name: 'Google.co.uk', value: 'google.co.uk' },
                    { name: 'Google.co.uz', value: 'google.co.uz' },
                    { name: 'Google.co.ve', value: 'google.co.ve' },
                    { name: 'Google.co.vi', value: 'google.co.vi' },
                    { name: 'Google.co.za', value: 'google.co.za' },
                    { name: 'Google.co.zw', value: 'google.co.zw' },
                    { name: 'Google.com', value: 'google.com' },
                    { name: 'Google.com.af', value: 'google.com.af' },
                    { name: 'Google.com.ag', value: 'google.com.ag' },
                    { name: 'Google.com.ai', value: 'google.com.ai' },
                    { name: 'Google.com.ar', value: 'google.com.ar' },
                    { name: 'Google.com.au', value: 'google.com.au' },
                    { name: 'Google.com.bd', value: 'google.com.bd' },
                    { name: 'Google.com.bh', value: 'google.com.bh' },
                    { name: 'Google.com.bn', value: 'google.com.bn' },
                    { name: 'Google.com.bo', value: 'google.com.bo' },
                    { name: 'Google.com.br', value: 'google.com.br' },
                    { name: 'Google.com.bz', value: 'google.com.bz' },
                    { name: 'Google.com.co', value: 'google.com.co' },
                    { name: 'Google.com.cu', value: 'google.com.cu' },
                    { name: 'Google.com.cy', value: 'google.com.cy' },
                    { name: 'Google.com.do', value: 'google.com.do' },
                    { name: 'Google.com.ec', value: 'google.com.ec' },
                    { name: 'Google.com.eg', value: 'google.com.eg' },
                    { name: 'Google.com.et', value: 'google.com.et' },
                    { name: 'Google.com.fj', value: 'google.com.fj' },
                    { name: 'Google.com.gh', value: 'google.com.gh' },
                    { name: 'Google.com.gi', value: 'google.com.gi' },
                    { name: 'Google.com.gt', value: 'google.com.gt' },
                    { name: 'Google.com.hk', value: 'google.com.hk' },
                    { name: 'Google.com.jm', value: 'google.com.jm' },
                    { name: 'Google.com.kh', value: 'google.com.kh' },
                    { name: 'Google.com.kn', value: 'google.com.kn' },
                    { name: 'Google.com.kw', value: 'google.com.kw' },
                    { name: 'Google.com.lb', value: 'google.com.lb' },
                    { name: 'Google.com.lc', value: 'google.com.lc' },
                    { name: 'Google.com.ly', value: 'google.com.ly' },
                    { name: 'Google.com.mm', value: 'google.com.mm' },
                    { name: 'Google.com.mt', value: 'google.com.mt' },
                    { name: 'Google.com.mx', value: 'google.com.mx' },
                    { name: 'Google.com.my', value: 'google.com.my' },
                    { name: 'Google.com.na', value: 'google.com.na' },
                    { name: 'Google.com.ng', value: 'google.com.ng' },
                    { name: 'Google.com.ni', value: 'google.com.ni' },
                    { name: 'Google.com.np', value: 'google.com.np' },
                    { name: 'Google.com.om', value: 'google.com.om' },
                    { name: 'Google.com.pa', value: 'google.com.pa' },
                    { name: 'Google.com.pe', value: 'google.com.pe' },
                    { name: 'Google.com.pg', value: 'google.com.pg' },
                    { name: 'Google.com.ph', value: 'google.com.ph' },
                    { name: 'Google.com.pk', value: 'google.com.pk' },
                    { name: 'Google.com.pr', value: 'google.com.pr' },
                    { name: 'Google.com.py', value: 'google.com.py' },
                    { name: 'Google.com.qa', value: 'google.com.qa' },
                    { name: 'Google.com.sa', value: 'google.com.sa' },
                    { name: 'Google.com.sb', value: 'google.com.sb' },
                    { name: 'Google.com.sg', value: 'google.com.sg' },
                    { name: 'Google.com.sl', value: 'google.com.sl' },
                    { name: 'Google.com.tj', value: 'google.com.tj' },
                    { name: 'Google.com.tr', value: 'google.com.tr' },
                    { name: 'Google.com.ua', value: 'google.com.ua' },
                    { name: 'Google.com.uy', value: 'google.com.uy' },
                    { name: 'Google.com.vc', value: 'google.com.vc' },
                    { name: 'Google.com.vg', value: 'google.com.vg' },
                    { name: 'Google.cv', value: 'google.cv' },
                    { name: 'Google.cz', value: 'google.cz' },
                    { name: 'Google.de', value: 'google.de' },
                    { name: 'Google.dj', value: 'google.dj' },
                    { name: 'Google.dk', value: 'google.dk' },
                    { name: 'Google.dm', value: 'google.dm' },
                    { name: 'Google.dz', value: 'google.dz' },
                    { name: 'Google.ee', value: 'google.ee' },
                    { name: 'Google.eh', value: 'google.eh' },
                    { name: 'Google.es', value: 'google.es' },
                    { name: 'Google.fi', value: 'google.fi' },
                    { name: 'Google.fm', value: 'google.fm' },
                    { name: 'Google.fr', value: 'google.fr' },
                    { name: 'Google.ga', value: 'google.ga' },
                    { name: 'Google.ge', value: 'google.ge' },
                    { name: 'Google.gg', value: 'google.gg' },
                    { name: 'Google.gl', value: 'google.gl' },
                    { name: 'Google.gm', value: 'google.gm' },
                    { name: 'Google.gp', value: 'google.gp' },
                    { name: 'Google.gr', value: 'google.gr' },
                    { name: 'Google.gs', value: 'google.gs' },
                    { name: 'Google.gy', value: 'google.gy' },
                    { name: 'Google.hn', value: 'google.hn' },
                    { name: 'Google.hr', value: 'google.hr' },
                    { name: 'Google.ht', value: 'google.ht' },
                    { name: 'Google.hu', value: 'google.hu' },
                    { name: 'Google.ie', value: 'google.ie' },
                    { name: 'Google.iq', value: 'google.iq' },
                    { name: 'Google.is', value: 'google.is' },
                    { name: 'Google.it', value: 'google.it' },
                    { name: 'Google.jo', value: 'google.jo' },
                    { name: 'Google.kg', value: 'google.kg' },
                    { name: 'Google.ki', value: 'google.ki' },
                    { name: 'Google.kz', value: 'google.kz' },
                    { name: 'Google.la', value: 'google.la' },
                    { name: 'Google.li', value: 'google.li' },
                    { name: 'Google.lk', value: 'google.lk' },
                    { name: 'Google.lt', value: 'google.lt' },
                    { name: 'Google.lu', value: 'google.lu' },
                    { name: 'Google.lv', value: 'google.lv' },
                    { name: 'Google.md', value: 'google.md' },
                    { name: 'Google.mg', value: 'google.mg' },
                    { name: 'Google.mk', value: 'google.mk' },
                    { name: 'Google.ml', value: 'google.ml' },
                    { name: 'Google.mn', value: 'google.mn' },
                    { name: 'Google.ms', value: 'google.ms' },
                    { name: 'Google.mu', value: 'google.mu' },
                    { name: 'Google.mv', value: 'google.mv' },
                    { name: 'Google.mw', value: 'google.mw' },
                    { name: 'Google.ne', value: 'google.ne' },
                    { name: 'Google.nl', value: 'google.nl' },
                    { name: 'Google.no', value: 'google.no' },
                    { name: 'Google.nr', value: 'google.nr' },
                    { name: 'Google.nu', value: 'google.nu' },
                    { name: 'Google.pl', value: 'google.pl' },
                    { name: 'Google.pm', value: 'google.pm' },
                    { name: 'Google.pn', value: 'google.pn' },
                    { name: 'Google.ps', value: 'google.ps' },
                    { name: 'Google.pt', value: 'google.pt' },
                    { name: 'Google.pw', value: 'google.pw' },
                    { name: 'Google.re', value: 'google.re' },
                    { name: 'Google.ro', value: 'google.ro' },
                    { name: 'Google.rs', value: 'google.rs' },
                    { name: 'Google.ru', value: 'google.ru' },
                    { name: 'Google.rw', value: 'google.rw' },
                    { name: 'Google.sc', value: 'google.sc' },
                    { name: 'Google.sd', value: 'google.sd' },
                    { name: 'Google.se', value: 'google.se' },
                    { name: 'Google.sh', value: 'google.sh' },
                    { name: 'Google.si', value: 'google.si' },
                    { name: 'Google.sj', value: 'google.sj' },
                    { name: 'Google.sk', value: 'google.sk' },
                    { name: 'Google.sm', value: 'google.sm' },
                    { name: 'Google.sn', value: 'google.sn' },
                    { name: 'Google.so', value: 'google.so' },
                    { name: 'Google.sr', value: 'google.sr' },
                    { name: 'Google.st', value: 'google.st' },
                    { name: 'Google.sy', value: 'google.sy' },
                    { name: 'Google.sz', value: 'google.sz' },
                    { name: 'Google.tc', value: 'google.tc' },
                    { name: 'Google.tg', value: 'google.tg' },
                    { name: 'Google.tk', value: 'google.tk' },
                    { name: 'Google.tl', value: 'google.tl' },
                    { name: 'Google.tm', value: 'google.tm' },
                    { name: 'Google.tn', value: 'google.tn' },
                    { name: 'Google.to', value: 'google.to' },
                    { name: 'Google.tt', value: 'google.tt' },
                    { name: 'Google.tv', value: 'google.tv' },
                    { name: 'Google.tw', value: 'google.tw' },
                    { name: 'Google.vn', value: 'google.vn' },
                    { name: 'Google.vu', value: 'google.vu' },
                    { name: 'Google.wf', value: 'google.wf' },
                    { name: 'Google.ws', value: 'google.ws' },
                    { name: 'Google.ye', value: 'google.ye' },
                    { name: 'Google.zm', value: 'google.zm' },
                ],
                routing: {
                    request: {
                        qs: {
                            google_domain: '={{$value}}',
                        },
                    },
                },
            },
        ],
        displayOptions,
    },
    {
        displayName: 'Language Settings',
        name: 'languageSettings',
        type: 'collection',
        placeholder: 'Add Language Setting',
        default: {},
        options: [
            {
                displayName: 'Interface Language',
                name: 'hl',
                type: 'options',
                default: 'en',
                description: 'The default parameter en defines the interface language of the search. Check the full list of supported Google hl languages.',
                options: [
                    { name: 'Afrikaans', value: 'af' },
                    { name: 'Akan', value: 'ak' },
                    { name: 'Albanian', value: 'sq' },
                    { name: 'Amharic', value: 'am' },
                    { name: 'Arabic', value: 'ar' },
                    { name: 'Armenian', value: 'hy' },
                    { name: 'Azerbaijani', value: 'az' },
                    { name: 'Basque', value: 'eu' },
                    { name: 'Belarusian', value: 'be' },
                    { name: 'Bemba', value: 'bem' },
                    { name: 'Bengali', value: 'bn' },
                    { name: 'Bihari', value: 'bh' },
                    { name: 'Bosnian', value: 'bs' },
                    { name: 'Breton', value: 'br' },
                    { name: 'Bulgarian', value: 'bg' },
                    { name: 'Cambodian', value: 'km' },
                    { name: 'Catalan', value: 'ca' },
                    { name: 'Cherokee', value: 'chr' },
                    { name: 'Chinese (Simplified)', value: 'zh-cn' },
                    { name: 'Chinese (Traditional)', value: 'zh-tw' },
                    { name: 'Corsican', value: 'co' },
                    { name: 'Croatian', value: 'hr' },
                    { name: 'Czech', value: 'cs' },
                    { name: 'Danish', value: 'da' },
                    { name: 'Dutch', value: 'nl' },
                    { name: 'English', value: 'en' },
                    { name: 'Esperanto', value: 'eo' },
                    { name: 'Estonian', value: 'et' },
                    { name: 'Ewe', value: 'ee' },
                    { name: 'Faroese', value: 'fo' },
                    { name: 'Filipino', value: 'tl' },
                    { name: 'Finnish', value: 'fi' },
                    { name: 'French', value: 'fr' },
                    { name: 'Frisian', value: 'fy' },
                    { name: 'Ga', value: 'gaa' },
                    { name: 'Galician', value: 'gl' },
                    { name: 'Georgian', value: 'ka' },
                    { name: 'German', value: 'de' },
                    { name: 'Greek', value: 'el' },
                    { name: 'Greenlandic', value: 'kl' },
                    { name: 'Guarani', value: 'gn' },
                    { name: 'Gujarati', value: 'gu' },
                    { name: 'Haitian Creole', value: 'ht' },
                    { name: 'Hausa', value: 'ha' },
                    { name: 'Hawaiian', value: 'haw' },
                    { name: 'Hebrew', value: 'iw' },
                    { name: 'Hindi', value: 'hi' },
                    { name: 'Hungarian', value: 'hu' },
                    { name: 'Icelandic', value: 'is' },
                    { name: 'Igbo', value: 'ig' },
                    { name: 'Indonesian', value: 'id' },
                    { name: 'Interlingua', value: 'ia' },
                    { name: 'Irish', value: 'ga' },
                    { name: 'Italian', value: 'it' },
                    { name: 'Japanese', value: 'ja' },
                    { name: 'Javanese', value: 'jw' },
                    { name: 'Kannada', value: 'kn' },
                    { name: 'Kazakh', value: 'kk' },
                    { name: 'Kinyarwanda', value: 'rw' },
                    { name: 'Kirundi', value: 'rn' },
                    { name: 'Kongo', value: 'kg' },
                    { name: 'Korean', value: 'ko' },
                    { name: 'Krio', value: 'kri' },
                    { name: 'Kurdish', value: 'ku' },
                    { name: 'Kurdish (Sorani)', value: 'ckb' },
                    { name: 'Kyrgyz', value: 'ky' },
                    { name: 'Laothian', value: 'lo' },
                    { name: 'Latin', value: 'la' },
                    { name: 'Latvian', value: 'lv' },
                    { name: 'Lingala', value: 'ln' },
                    { name: 'Lithuanian', value: 'lt' },
                    { name: 'Lozi', value: 'loz' },
                    { name: 'Luganda', value: 'lg' },
                    { name: 'Luo', value: 'ach' },
                    { name: 'Macedonian', value: 'mk' },
                    { name: 'Malagasy', value: 'mg' },
                    { name: 'Malay', value: 'ms' },
                    { name: 'Malayalam', value: 'ml' },
                    { name: 'Maltese', value: 'mt' },
                    { name: 'Maori', value: 'mi' },
                    { name: 'Marathi', value: 'mr' },
                    { name: 'Mauritian Creole', value: 'mfe' },
                    { name: 'Moldavian', value: 'mo' },
                    { name: 'Mongolian', value: 'mn' },
                    { name: 'Montenegrin', value: 'sr-me' },
                    { name: 'Myanmar', value: 'my' },
                    { name: 'Nepali', value: 'ne' },
                    { name: 'Nigerian Pidgin', value: 'pcm' },
                    { name: 'Northern Sotho', value: 'nso' },
                    { name: 'Norwegian', value: 'no' },
                    { name: 'Norwegian (Nynorsk)', value: 'nn' },
                    { name: 'Nyanja', value: 'ny' },
                    { name: 'Occitan', value: 'oc' },
                    { name: 'Oriya', value: 'or' },
                    { name: 'Oromo', value: 'om' },
                    { name: 'Pashto', value: 'ps' },
                    { name: 'Persian', value: 'fa' },
                    { name: 'Polish', value: 'pl' },
                    { name: 'Portuguese', value: 'pt' },
                    { name: 'Portuguese (Brazil)', value: 'pt-br' },
                    { name: 'Portuguese (Portugal)', value: 'pt-pt' },
                    { name: 'Punjabi', value: 'pa' },
                    { name: 'Quechua', value: 'qu' },
                    { name: 'Romanian', value: 'ro' },
                    { name: 'Romansh', value: 'rm' },
                    { name: 'Runyakitara', value: 'nyn' },
                    { name: 'Russian', value: 'ru' },
                    { name: 'Scots Gaelic', value: 'gd' },
                    { name: 'Serbian', value: 'sr' },
                    { name: 'Serbo-Croatian', value: 'sh' },
                    { name: 'Sesotho', value: 'st' },
                    { name: 'Setswana', value: 'tn' },
                    { name: 'Seychellois Creole', value: 'crs' },
                    { name: 'Shona', value: 'sn' },
                    { name: 'Sindhi', value: 'sd' },
                    { name: 'Sinhalese', value: 'si' },
                    { name: 'Slovak', value: 'sk' },
                    { name: 'Slovenian', value: 'sl' },
                    { name: 'Somali', value: 'so' },
                    { name: 'Spanish', value: 'es' },
                    { name: 'Spanish (Latin America)', value: 'es-419' },
                    { name: 'Sundanese', value: 'su' },
                    { name: 'Swahili', value: 'sw' },
                    { name: 'Swedish', value: 'sv' },
                    { name: 'Tajik', value: 'tg' },
                    { name: 'Tamil', value: 'ta' },
                    { name: 'Tatar', value: 'tt' },
                    { name: 'Telugu', value: 'te' },
                    { name: 'Thai', value: 'th' },
                    { name: 'Tigrinya', value: 'ti' },
                    { name: 'Tonga', value: 'to' },
                    { name: 'Tshiluba', value: 'lua' },
                    { name: 'Tumbuka', value: 'tum' },
                    { name: 'Turkish', value: 'tr' },
                    { name: 'Turkmen', value: 'tk' },
                    { name: 'Twi', value: 'tw' },
                    { name: 'Uighur', value: 'ug' },
                    { name: 'Ukrainian', value: 'uk' },
                    { name: 'Urdu', value: 'ur' },
                    { name: 'Uzbek', value: 'uz' },
                    { name: 'Vietnamese', value: 'vi' },
                    { name: 'Welsh', value: 'cy' },
                    { name: 'Wolof', value: 'wo' },
                    { name: 'Xhosa', value: 'xh' },
                    { name: 'Yiddish', value: 'yi' },
                    { name: 'Yoruba', value: 'yo' },
                    { name: 'Zulu', value: 'zu' },
                ],
                routing: {
                    request: {
                        qs: {
                            hl: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Language Restriction',
                name: 'lr',
                type: 'multiOptions',
                default: ['lang_en'],
                description: 'The lr parameter restricts search results to documents written in a particular language or a set of languages. You can select multiple languages to include documents written in any of those languages.',
                options: [
                    { name: 'Arabic', value: 'lang_ar' },
                    { name: 'Armenian', value: 'lang_hy' },
                    { name: 'Bulgarian', value: 'lang_bg' },
                    { name: 'Catalan', value: 'lang_ca' },
                    { name: 'Chinese (Simplified)', value: 'lang_zh-CN' },
                    { name: 'Chinese (Traditional)', value: 'lang_zh-TW' },
                    { name: 'Croatian', value: 'lang_hr' },
                    { name: 'Czech', value: 'lang_cs' },
                    { name: 'Danish', value: 'lang_da' },
                    { name: 'Dutch', value: 'lang_nl' },
                    { name: 'English', value: 'lang_en' },
                    { name: 'Estonian', value: 'lang_et' },
                    { name: 'Filipino', value: 'lang_tl' },
                    { name: 'Finnish', value: 'lang_fi' },
                    { name: 'French', value: 'lang_fr' },
                    { name: 'German', value: 'lang_de' },
                    { name: 'Greek', value: 'lang_el' },
                    { name: 'Hebrew', value: 'lang_iw' },
                    { name: 'Hindi', value: 'lang_hi' },
                    { name: 'Hungarian', value: 'lang_hu' },
                    { name: 'Icelandic', value: 'lang_is' },
                    { name: 'Indonesian', value: 'lang_id' },
                    { name: 'Italian', value: 'lang_it' },
                    { name: 'Japanese', value: 'lang_ja' },
                    { name: 'Korean', value: 'lang_ko' },
                    { name: 'Latvian', value: 'lang_lv' },
                    { name: 'Lithuanian', value: 'lang_lt' },
                    { name: 'Norwegian', value: 'lang_no' },
                    { name: 'Persian', value: 'lang_fa' },
                    { name: 'Polish', value: 'lang_pl' },
                    { name: 'Portuguese', value: 'lang_pt' },
                    { name: 'Romanian', value: 'lang_ro' },
                    { name: 'Russian', value: 'lang_ru' },
                    { name: 'Serbian', value: 'lang_sr' },
                    { name: 'Slovak', value: 'lang_sk' },
                    { name: 'Slovenian', value: 'lang_sl' },
                    { name: 'Spanish', value: 'lang_es' },
                    { name: 'Swedish', value: 'lang_sv' },
                    { name: 'Thai', value: 'lang_th' },
                    { name: 'Turkish', value: 'lang_tr' },
                    { name: 'Ukrainian', value: 'lang_uk' },
                    { name: 'Vietnamese', value: 'lang_vi' },
                ],
                routing: {
                    request: {
                        qs: {
                            lr: '={{$value.join("|")}}',
                        },
                    },
                },
            },
        ],
        displayOptions,
    },
    {
        displayName: 'Search Options',
        name: 'searchOptions',
        type: 'collection',
        placeholder: 'Add Search Option',
        default: {},
        options: [
            {
                displayName: 'Content Filter',
                name: 'filter',
                type: 'options',
                default: '1',
                description: 'This parameter controls whether the \'Duplicate Content\' and \'Host Crowding\' filters are enabled. Set the value to 1 to enable these filters, which is the default setting. To disable these filters, set the value to 0.',
                options: [
                    { name: 'Disable Filters', value: '0' },
                    { name: 'Enable Filters', value: '1' },
                ],
                routing: {
                    request: {
                        qs: {
                            filter: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Device Type',
                name: 'device',
                type: 'options',
                default: 'desktop',
                description: 'The default parameter desktop defines the search on a desktop device. The mobile parameter defines the search on a mobile device. The tablet parameter defines the search on a tablet device.',
                options: [
                    { name: 'Desktop', value: 'desktop' },
                    { name: 'Mobile', value: 'mobile' },
                    { name: 'Tablet', value: 'tablet' },
                ],
                routing: {
                    request: {
                        qs: {
                            device: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'No Auto-Correction',
                name: 'nfpr',
                type: 'options',
                default: '0',
                description: 'This parameter controls whether results from queries that have been auto-corrected for spelling errors are included. To exclude these auto-corrected results, set the value to 1. By default, the value is 0, meaning auto-corrected results are included.',
                options: [
                    { name: 'Include Auto-Corrected Results', value: '0' },
                    { name: 'Exclude Auto-Corrected Results', value: '1' },
                ],
                routing: {
                    request: {
                        qs: {
                            nfpr: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Optimization Strategy',
                name: 'optimization_strategy',
                type: 'options',
                default: 'performance',
                description: 'Controls how the search request is optimized. Available options: performance (Default), ads - optimizes for higher ad collection success rate, which may result in longer request processing times.',
                options: [
                    { name: 'Ads', value: 'ads' },
                    { name: 'Performance', value: 'performance' },
                ],
                routing: {
                    request: {
                        qs: {
                            optimization_strategy: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'SafeSearch',
                name: 'safe',
                type: 'options',
                default: 'off',
                description: 'This parameter toggles the SafeSearch feature for the results. SafeSearch operates by filtering out adult content from your search results. Google\'s filters use proprietary technology to check keywords, phrases and URLs. While no filters are 100 percent accurate, SafeSearch will remove the overwhelming majority of adult content from your search results. Set the value to active to enable SafeSearch. To disable it, set the value to off. By default, SafeSearch is disabled.',
                options: [
                    { name: 'Active', value: 'active' },
                    { name: 'Off', value: 'off' },
                ],
                routing: {
                    request: {
                        qs: {
                            safe: '={{$value}}',
                        },
                    },
                },
            },
        ],
        displayOptions,
    },
    {
        displayName: 'Time Filters',
        name: 'timeFilters',
        type: 'collection',
        placeholder: 'Add Time Filter',
        default: {},
        options: [
            {
                displayName: 'Time Period',
                name: 'time_period',
                type: 'options',
                default: 'last_day',
                description: 'This parameter restricts results to URLs based on date. Supported values are: last_hour - data from the past hour. last_day - data from the past 24 hours. last_week - data from the past week. last_month - data from the past month. last_year - data from the past year. Using time_period_min or time_period_max parameters, you can specify a custom time period. Note, that the time_period_min and time_period_max parameters could be used separately as well.',
                options: [
                    { name: 'Last Day', value: 'last_day' },
                    { name: 'Last Hour', value: 'last_hour' },
                    { name: 'Last Month', value: 'last_month' },
                    { name: 'Last Week', value: 'last_week' },
                    { name: 'Last Year', value: 'last_year' },
                ],
                routing: {
                    request: {
                        qs: {
                            time_period: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Time Period Start',
                name: 'time_period_min',
                type: 'string',
                default: '',
                description: 'This parameter specifies the start of the time period. It could be used in combination with the time_period_max parameter. The value should be in the format MM/DD/YYYY.',
                routing: {
                    request: {
                        qs: {
                            time_period_min: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Time Period End',
                name: 'time_period_max',
                type: 'string',
                default: '',
                description: 'This parameter specifies the end of the time period. It could be used in combination with the time_period_min parameter. The value should be in the format MM/DD/YYYY.',
                routing: {
                    request: {
                        qs: {
                            time_period_max: '={{$value}}',
                        },
                    },
                },
            },
        ],
        displayOptions,
    },
    {
        displayName: 'Pagination',
        name: 'pagination',
        type: 'collection',
        placeholder: 'Add Pagination Setting',
        default: {},
        options: [
            {
                displayName: 'Results Per Page',
                name: 'num',
                type: 'string',
                default: '10',
                description: 'This parameter specifies the number of results to display per page. Use in combination with the page parameter to implement pagination functionality.',
                routing: {
                    request: {
                        qs: {
                            num: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Page Number',
                name: 'page',
                type: 'string',
                default: '1',
                description: 'This parameter indicates which page of results to return. By default, it is set to 1. Use in combination with the num parameter to implement pagination.',
                routing: {
                    request: {
                        qs: {
                            page: '={{$value}}',
                        },
                    },
                },
            },
        ],
        displayOptions,
    },
    {
        displayName: 'Advanced Options',
        name: 'advancedOptions',
        type: 'collection',
        placeholder: 'Add Advanced Option',
        default: {},
        options: [
            {
                displayName: 'Knowledge Graph ID',
                name: 'kgmid',
                type: 'string',
                default: '',
                description: 'Defines a Knowledge Graph identifier (kgmid), representing entities in Google\'s Knowledge Graph. Format: Location Identifier (/m/): Typically followed by 2 to 7 characters. Used primarily to represent specific locations. Find the identifier by searching for the \'Freebase ID\' on Wikidata. Example: kgmid=/m/02_286 refers to New York. Google Knowledge Graph Identifier (/g/): Typically followed by a longer alphanumeric string. Represents general entities in Google\'s Knowledge Graph. Find details on Wikidata. Example: kgmid=/g/11f555cn8l refers to TikTok.',
                routing: {
                    request: {
                        qs: {
                            kgmid: '={{$value}}',
                        },
                    },
                },
            },
        ],
        displayOptions,
    },
];
exports.google = {
    resource,
    properties,
};
