// @searchapi/n8n-nodes-searchapi/dist/nodes/SearchApi/engines/google_images.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.google_images = void 0;
const displayOptions = {
    show: {
        resource: ['google_images'],
    },
};
const resource = {
    name: 'Google Images',
    value: 'google_images'
};
const properties = [
    {
        displayName: 'Query',
        name: 'q',
        type: 'string',
        required: true,
        default: '',
        description: 'Parameter defines the query you want to search. You can use anything that you would use in a regular Google search. e.g. inurl:, site:, intitle:. We also support advanced search query parameters such as as_dt and as_eq.',
        options: [],
        displayOptions,
        routing: {
            request: {
                qs: {
                    q: '={{$value}}',
                },
            },
        },
    },
    {
        displayName: 'Location & Language Settings',
        name: 'locationLanguageSettings',
        type: 'collection',
        placeholder: 'Add Location/Language Setting',
        default: {},
        options: [
            {
                displayName: 'Geographic Location',
                name: 'location',
                type: 'string',
                default: '',
                description: 'Parameter defines from where you want the search to originate. If several locations match the location requested, we\'ll pick the most popular one. Head to the Locations API if you need more precise control.',
                routing: {
                    request: {
                        qs: {
                            location: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Encoded Location',
                name: 'uule',
                type: 'string',
                default: '',
                description: 'Parameter is the Google encoded location you want to use for the search. SearchApi automatically generated the uule parameter when you use the location parameter but we allow you to overwrite it directly. uule and location parameters can\'t be used together.',
                routing: {
                    request: {
                        qs: {
                            uule: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Country',
                name: 'gl',
                type: 'options',
                default: 'us',
                description: 'The default parameter us defines the country of the search. Check the full list of supported Google gl countries.',
                options: [
                    { name: 'Afghanistan', value: 'af' },
                    { name: 'Albania', value: 'al' },
                    { name: 'Algeria', value: 'dz' },
                    { name: 'American Samoa', value: 'as' },
                    { name: 'Andorra', value: 'ad' },
                    { name: 'Angola', value: 'ao' },
                    { name: 'Anguilla', value: 'ai' },
                    { name: 'Antarctica', value: 'aq' },
                    { name: 'Antigua and Barbuda', value: 'ag' },
                    { name: 'Argentina', value: 'ar' },
                    { name: 'Armenia', value: 'am' },
                    { name: 'Aruba', value: 'aw' },
                    { name: 'Australia', value: 'au' },
                    { name: 'Austria', value: 'at' },
                    { name: 'Azerbaijan', value: 'az' },
                    { name: 'Bahamas', value: 'bs' },
                    { name: 'Bahrain', value: 'bh' },
                    { name: 'Bangladesh', value: 'bd' },
                    { name: 'Barbados', value: 'bb' },
                    { name: 'Belarus', value: 'by' },
                    { name: 'Belgium', value: 'be' },
                    { name: 'Belize', value: 'bz' },
                    { name: 'Benin', value: 'bj' },
                    { name: 'Bermuda', value: 'bm' },
                    { name: 'Bhutan', value: 'bt' },
                    { name: 'Bolivia', value: 'bo' },
                    { name: 'Bosnia and Herzegovina', value: 'ba' },
                    { name: 'Botswana', value: 'bw' },
                    { name: 'Bouvet Island', value: 'bv' },
                    { name: 'Brazil', value: 'br' },
                    { name: 'British Indian Ocean Territory', value: 'io' },
                    { name: 'Brunei', value: 'bn' },
                    { name: 'Bulgaria', value: 'bg' },
                    { name: 'Burkina Faso', value: 'bf' },
                    { name: 'Burundi', value: 'bi' },
                    { name: 'Cambodia', value: 'kh' },
                    { name: 'Cameroon', value: 'cm' },
                    { name: 'Canada', value: 'ca' },
                    { name: 'Cape Verde', value: 'cv' },
                    { name: 'Cayman Islands', value: 'ky' },
                    { name: 'Central African Republic', value: 'cf' },
                    { name: 'Chad', value: 'td' },
                    { name: 'Chile', value: 'cl' },
                    { name: 'China', value: 'cn' },
                    { name: 'Christmas Island', value: 'cx' },
                    { name: 'Cocos Islands', value: 'cc' },
                    { name: 'Colombia', value: 'co' },
                    { name: 'Comoros', value: 'km' },
                    { name: 'Congo', value: 'cg' },
                    { name: 'Congo (Democratic Republic)', value: 'cd' },
                    { name: 'Cook Islands', value: 'ck' },
                    { name: 'Costa Rica', value: 'cr' },
                    { name: 'Cote d\'Ivoire', value: 'ci' },
                    { name: 'Croatia', value: 'hr' },
                    { name: 'Cuba', value: 'cu' },
                    { name: 'Cyprus', value: 'cy' },
                    { name: 'Czech Republic', value: 'cz' },
                    { name: 'Denmark', value: 'dk' },
                    { name: 'Djibouti', value: 'dj' },
                    { name: 'Dominica', value: 'dm' },
                    { name: 'Dominican Republic', value: 'do' },
                    { name: 'Ecuador', value: 'ec' },
                    { name: 'Egypt', value: 'eg' },
                    { name: 'El Salvador', value: 'sv' },
                    { name: 'Equatorial Guinea', value: 'gq' },
                    { name: 'Eritrea', value: 'er' },
                    { name: 'Estonia', value: 'ee' },
                    { name: 'Ethiopia', value: 'et' },
                    { name: 'Falkland Islands', value: 'fk' },
                    { name: 'Faroe Islands', value: 'fo' },
                    { name: 'Fiji', value: 'fj' },
                    { name: 'Finland', value: 'fi' },
                    { name: 'France', value: 'fr' },
                    { name: 'French Guiana', value: 'gf' },
                    { name: 'French Polynesia', value: 'pf' },
                    { name: 'French Southern Territories', value: 'tf' },
                    { name: 'Gabon', value: 'ga' },
                    { name: 'Gambia', value: 'gm' },
                    { name: 'Georgia', value: 'ge' },
                    { name: 'Germany', value: 'de' },
                    { name: 'Ghana', value: 'gh' },
                    { name: 'Gibraltar', value: 'gi' },
                    { name: 'Great Britain', value: 'gb' },
                    { name: 'Greece', value: 'gr' },
                    { name: 'Greenland', value: 'gl' },
                    { name: 'Grenada', value: 'gd' },
                    { name: 'Guadeloupe', value: 'gp' },
                    { name: 'Guam', value: 'gu' },
                    { name: 'Guatemala', value: 'gt' },
                    { name: 'Guernsey', value: 'gg' },
                    { name: 'Guinea', value: 'gn' },
                    { name: 'Guinea-Bissau', value: 'gw' },
                    { name: 'Guyana', value: 'gy' },
                    { name: 'Haiti', value: 'ht' },
                    { name: 'Heard Island and McDonald Islands', value: 'hm' },
                    { name: 'Holy See', value: 'va' },
                    { name: 'Honduras', value: 'hn' },
                    { name: 'Hong Kong', value: 'hk' },
                    { name: 'Hungary', value: 'hu' },
                    { name: 'Iceland', value: 'is' },
                    { name: 'India', value: 'in' },
                    { name: 'Indonesia', value: 'id' },
                    { name: 'Iran', value: 'ir' },
                    { name: 'Iraq', value: 'iq' },
                    { name: 'Ireland', value: 'ie' },
                    { name: 'Isle of Man', value: 'im' },
                    { name: 'Israel', value: 'il' },
                    { name: 'Italy', value: 'it' },
                    { name: 'Jamaica', value: 'jm' },
                    { name: 'Japan', value: 'jp' },
                    { name: 'Jersey', value: 'je' },
                    { name: 'Jordan', value: 'jo' },
                    { name: 'Kazakhstan', value: 'kz' },
                    { name: 'Kenya', value: 'ke' },
                    { name: 'Kiribati', value: 'ki' },
                    { name: 'Korea (North)', value: 'kp' },
                    { name: 'Korea (South)', value: 'kr' },
                    { name: 'Kuwait', value: 'kw' },
                    { name: 'Kyrgyzstan', value: 'kg' },
                    { name: 'Laos', value: 'la' },
                    { name: 'Latvia', value: 'lv' },
                    { name: 'Lebanon', value: 'lb' },
                    { name: 'Lesotho', value: 'ls' },
                    { name: 'Liberia', value: 'lr' },
                    { name: 'Libya', value: 'ly' },
                    { name: 'Liechtenstein', value: 'li' },
                    { name: 'Lithuania', value: 'lt' },
                    { name: 'Luxembourg', value: 'lu' },
                    { name: 'Macao', value: 'mo' },
                    { name: 'Macedonia', value: 'mk' },
                    { name: 'Madagascar', value: 'mg' },
                    { name: 'Malawi', value: 'mw' },
                    { name: 'Malaysia', value: 'my' },
                    { name: 'Maldives', value: 'mv' },
                    { name: 'Mali', value: 'ml' },
                    { name: 'Malta', value: 'mt' },
                    { name: 'Marshall Islands', value: 'mh' },
                    { name: 'Martinique', value: 'mq' },
                    { name: 'Mauritania', value: 'mr' },
                    { name: 'Mauritius', value: 'mu' },
                    { name: 'Mayotte', value: 'yt' },
                    { name: 'Mexico', value: 'mx' },
                    { name: 'Micronesia', value: 'fm' },
                    { name: 'Moldova', value: 'md' },
                    { name: 'Monaco', value: 'mc' },
                    { name: 'Mongolia', value: 'mn' },
                    { name: 'Montenegro', value: 'me' },
                    { name: 'Montserrat', value: 'ms' },
                    { name: 'Morocco', value: 'ma' },
                    { name: 'Mozambique', value: 'mz' },
                    { name: 'Myanmar', value: 'mm' },
                    { name: 'Namibia', value: 'na' },
                    { name: 'Nauru', value: 'nr' },
                    { name: 'Nepal', value: 'np' },
                    { name: 'Netherlands', value: 'nl' },
                    { name: 'New Caledonia', value: 'nc' },
                    { name: 'New Zealand', value: 'nz' },
                    { name: 'Nicaragua', value: 'ni' },
                    { name: 'Niger', value: 'ne' },
                    { name: 'Nigeria', value: 'ng' },
                    { name: 'Niue', value: 'nu' },
                    { name: 'Norfolk Island', value: 'nf' },
                    { name: 'Northern Mariana Islands', value: 'mp' },
                    { name: 'Norway', value: 'no' },
                    { name: 'Oman', value: 'om' },
                    { name: 'Pakistan', value: 'pk' },
                    { name: 'Palau', value: 'pw' },
                    { name: 'Palestine', value: 'ps' },
                    { name: 'Panama', value: 'pa' },
                    { name: 'Papua New Guinea', value: 'pg' },
                    { name: 'Paraguay', value: 'py' },
                    { name: 'Peru', value: 'pe' },
                    { name: 'Philippines', value: 'ph' },
                    { name: 'Pitcairn', value: 'pn' },
                    { name: 'Poland', value: 'pl' },
                    { name: 'Portugal', value: 'pt' },
                    { name: 'Puerto Rico', value: 'pr' },
                    { name: 'Qatar', value: 'qa' },
                    { name: 'Reunion', value: 're' },
                    { name: 'Romania', value: 'ro' },
                    { name: 'Russia', value: 'ru' },
                    { name: 'Rwanda', value: 'rw' },
                    { name: 'Saint Helena', value: 'sh' },
                    { name: 'Saint Kitts and Nevis', value: 'kn' },
                    { name: 'Saint Lucia', value: 'lc' },
                    { name: 'Saint Pierre and Miquelon', value: 'pm' },
                    { name: 'Saint Vincent and the Grenadines', value: 'vc' },
                    { name: 'Samoa', value: 'ws' },
                    { name: 'San Marino', value: 'sm' },
                    { name: 'Sao Tome and Principe', value: 'st' },
                    { name: 'Saudi Arabia', value: 'sa' },
                    { name: 'Senegal', value: 'sn' },
                    { name: 'Serbia', value: 'rs' },
                    { name: 'Seychelles', value: 'sc' },
                    { name: 'Sierra Leone', value: 'sl' },
                    { name: 'Singapore', value: 'sg' },
                    { name: 'Slovakia', value: 'sk' },
                    { name: 'Slovenia', value: 'si' },
                    { name: 'Solomon Islands', value: 'sb' },
                    { name: 'Somalia', value: 'so' },
                    { name: 'South Africa', value: 'za' },
                    { name: 'South Georgia and the South Sandwich Islands', value: 'gs' },
                    { name: 'Spain', value: 'es' },
                    { name: 'Sri Lanka', value: 'lk' },
                    { name: 'Sudan', value: 'sd' },
                    { name: 'Suriname', value: 'sr' },
                    { name: 'Svalbard and Jan Mayen', value: 'sj' },
                    { name: 'Swaziland', value: 'sz' },
                    { name: 'Sweden', value: 'se' },
                    { name: 'Switzerland', value: 'ch' },
                    { name: 'Syria', value: 'sy' },
                    { name: 'Taiwan', value: 'tw' },
                    { name: 'Tajikistan', value: 'tj' },
                    { name: 'Tanzania', value: 'tz' },
                    { name: 'Thailand', value: 'th' },
                    { name: 'Timor-Leste', value: 'tl' },
                    { name: 'Togo', value: 'tg' },
                    { name: 'Tokelau', value: 'tk' },
                    { name: 'Tonga', value: 'to' },
                    { name: 'Trinidad and Tobago', value: 'tt' },
                    { name: 'Tunisia', value: 'tn' },
                    { name: 'Turkey', value: 'tr' },
                    { name: 'Turkmenistan', value: 'tm' },
                    { name: 'Turks and Caicos Islands', value: 'tc' },
                    { name: 'Tuvalu', value: 'tv' },
                    { name: 'Uganda', value: 'ug' },
                    { name: 'Ukraine', value: 'ua' },
                    { name: 'United Arab Emirates', value: 'ae' },
                    { name: 'United Kingdom', value: 'uk' },
                    { name: 'United States', value: 'us' },
                    { name: 'United States Minor Outlying Islands', value: 'um' },
                    { name: 'Uruguay', value: 'uy' },
                    { name: 'Uzbekistan', value: 'uz' },
                    { name: 'Vanuatu', value: 'vu' },
                    { name: 'Venezuela', value: 've' },
                    { name: 'Vietnam', value: 'vn' },
                    { name: 'Virgin Islands (British)', value: 'vg' },
                    { name: 'Virgin Islands (U.S.)', value: 'vi' },
                    { name: 'Wallis and Futuna', value: 'wf' },
                    { name: 'Western Sahara', value: 'eh' },
                    { name: 'Yemen', value: 'ye' },
                    { name: 'Zambia', value: 'zm' },
                    { name: 'Zimbabwe', value: 'zw' },
                ],
                routing: {
                    request: {
                        qs: {
                            gl: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Interface Language',
                name: 'hl',
                type: 'options',
                default: 'en',
                description: 'The default parameter en defines the interface language of the search. Check the full list of supported Google hl languages.',
                options: [
                    { name: 'Afrikaans', value: 'af' },
                    { name: 'Akan', value: 'ak' },
                    { name: 'Albanian', value: 'sq' },
                    { name: 'Amharic', value: 'am' },
                    { name: 'Arabic', value: 'ar' },
                    { name: 'Armenian', value: 'hy' },
                    { name: 'Azerbaijani', value: 'az' },
                    { name: 'Basque', value: 'eu' },
                    { name: 'Belarusian', value: 'be' },
                    { name: 'Bemba', value: 'bem' },
                    { name: 'Bengali', value: 'bn' },
                    { name: 'Bihari', value: 'bh' },
                    { name: 'Bork', value: 'xx-bork' },
                    { name: 'Bosnian', value: 'bs' },
                    { name: 'Breton', value: 'br' },
                    { name: 'Bulgarian', value: 'bg' },
                    { name: 'Cambodian', value: 'km' },
                    { name: 'Catalan', value: 'ca' },
                    { name: 'Cherokee', value: 'chr' },
                    { name: 'Chichewa', value: 'ny' },
                    { name: 'Chinese (Simplified)', value: 'zh-cn' },
                    { name: 'Chinese (Traditional)', value: 'zh-tw' },
                    { name: 'Corsican', value: 'co' },
                    { name: 'Croatian', value: 'hr' },
                    { name: 'Czech', value: 'cs' },
                    { name: 'Danish', value: 'da' },
                    { name: 'Dutch', value: 'nl' },
                    { name: 'Elmer Fudd', value: 'xx-elmer' },
                    { name: 'English', value: 'en' },
                    { name: 'Esperanto', value: 'eo' },
                    { name: 'Estonian', value: 'et' },
                    { name: 'Ewe', value: 'ee' },
                    { name: 'Faroese', value: 'fo' },
                    { name: 'Filipino', value: 'tl' },
                    { name: 'Finnish', value: 'fi' },
                    { name: 'French', value: 'fr' },
                    { name: 'Frisian', value: 'fy' },
                    { name: 'Gaa', value: 'gaa' },
                    { name: 'Galician', value: 'gl' },
                    { name: 'Georgian', value: 'ka' },
                    { name: 'German', value: 'de' },
                    { name: 'Greek', value: 'el' },
                    { name: 'Greenlandic', value: 'kl' },
                    { name: 'Guarani', value: 'gn' },
                    { name: 'Gujarati', value: 'gu' },
                    { name: 'Hacker', value: 'xx-hacker' },
                    { name: 'Haitian Creole', value: 'ht' },
                    { name: 'Hausa', value: 'ha' },
                    { name: 'Hawaiian', value: 'haw' },
                    { name: 'Hebrew', value: 'iw' },
                    { name: 'Hindi', value: 'hi' },
                    { name: 'Hungarian', value: 'hu' },
                    { name: 'Icelandic', value: 'is' },
                    { name: 'Igbo', value: 'ig' },
                    { name: 'Indonesian', value: 'id' },
                    { name: 'Interlingua', value: 'ia' },
                    { name: 'Irish', value: 'ga' },
                    { name: 'Italian', value: 'it' },
                    { name: 'Japanese', value: 'ja' },
                    { name: 'Javanese', value: 'jw' },
                    { name: 'Kannada', value: 'kn' },
                    { name: 'Kazakh', value: 'kk' },
                    { name: 'Kinyarwanda', value: 'rw' },
                    { name: 'Kirundi', value: 'rn' },
                    { name: 'Klingon', value: 'xx-klingon' },
                    { name: 'Kongo', value: 'kg' },
                    { name: 'Korean', value: 'ko' },
                    { name: 'Krio', value: 'kri' },
                    { name: 'Kurdish', value: 'ku' },
                    { name: 'Kurdish (Sorani)', value: 'ckb' },
                    { name: 'Kyrgyz', value: 'ky' },
                    { name: 'Laothian', value: 'lo' },
                    { name: 'Latin', value: 'la' },
                    { name: 'Latvian', value: 'lv' },
                    { name: 'Lingala', value: 'ln' },
                    { name: 'Lithuanian', value: 'lt' },
                    { name: 'Lozi', value: 'loz' },
                    { name: 'Luganda', value: 'lg' },
                    { name: 'Luo', value: 'ach' },
                    { name: 'Macedonian', value: 'mk' },
                    { name: 'Malagasy', value: 'mg' },
                    { name: 'Malay', value: 'my' },
                    { name: 'Malayalam', value: 'ml' },
                    { name: 'Maltese', value: 'mt' },
                    { name: 'Maori', value: 'mi' },
                    { name: 'Marathi', value: 'mr' },
                    { name: 'Mauritian Creole', value: 'mfe' },
                    { name: 'Moldavian', value: 'mo' },
                    { name: 'Mongolian', value: 'mn' },
                    { name: 'Montenegrin', value: 'sr-me' },
                    { name: 'Nepali', value: 'ne' },
                    { name: 'Nigerian Pidgin', value: 'pcm' },
                    { name: 'Northern Sotho', value: 'nso' },
                    { name: 'Norwegian', value: 'no' },
                    { name: 'Norwegian (Nynorsk)', value: 'nn' },
                    { name: 'Occitan', value: 'oc' },
                    { name: 'Oriya', value: 'or' },
                    { name: 'Oromo', value: 'om' },
                    { name: 'Pashto', value: 'ps' },
                    { name: 'Persian', value: 'fa' },
                    { name: 'Pirate', value: 'xx-pirate' },
                    { name: 'Polish', value: 'pl' },
                    { name: 'Portuguese', value: 'pt' },
                    { name: 'Portuguese (Brazil)', value: 'pt-br' },
                    { name: 'Portuguese (Portugal)', value: 'pt-pt' },
                    { name: 'Punjabi', value: 'pa' },
                    { name: 'Quechua', value: 'qu' },
                    { name: 'Romanian', value: 'ro' },
                    { name: 'Romansh', value: 'rm' },
                    { name: 'Runyakitara', value: 'nyn' },
                    { name: 'Russian', value: 'ru' },
                    { name: 'Scots Gaelic', value: 'gd' },
                    { name: 'Serbian', value: 'sr' },
                    { name: 'Serbo-Croatian', value: 'sh' },
                    { name: 'Sesotho', value: 'st' },
                    { name: 'Setswana', value: 'tn' },
                    { name: 'Seychellois Creole', value: 'crs' },
                    { name: 'Shona', value: 'sn' },
                    { name: 'Sindhi', value: 'sd' },
                    { name: 'Sinhalese', value: 'si' },
                    { name: 'Slovak', value: 'sk' },
                    { name: 'Slovenian', value: 'sl' },
                    { name: 'Somali', value: 'so' },
                    { name: 'Spanish', value: 'es' },
                    { name: 'Spanish (Latin America)', value: 'es-419' },
                    { name: 'Sundanese', value: 'su' },
                    { name: 'Swahili', value: 'sw' },
                    { name: 'Swedish', value: 'sv' },
                    { name: 'Tajik', value: 'tg' },
                    { name: 'Tamil', value: 'ta' },
                    { name: 'Tatar', value: 'tt' },
                    { name: 'Telugu', value: 'te' },
                    { name: 'Thai', value: 'th' },
                    { name: 'Tigrinya', value: 'ti' },
                    { name: 'Tonga', value: 'to' },
                    { name: 'Tshiluba', value: 'lua' },
                    { name: 'Tumbuka', value: 'tum' },
                    { name: 'Turkish', value: 'tr' },
                    { name: 'Turkmen', value: 'tk' },
                    { name: 'Twi', value: 'tw' },
                    { name: 'Uighur', value: 'ug' },
                    { name: 'Ukrainian', value: 'uk' },
                    { name: 'Urdu', value: 'ur' },
                    { name: 'Uzbek', value: 'uz' },
                    { name: 'Vietnamese', value: 'vi' },
                    { name: 'Welsh', value: 'cy' },
                    { name: 'Wolof', value: 'wo' },
                    { name: 'Xhosa', value: 'xh' },
                    { name: 'Yiddish', value: 'yi' },
                    { name: 'Yoruba', value: 'yo' },
                    { name: 'Zulu', value: 'zu' },
                ],
                routing: {
                    request: {
                        qs: {
                            hl: '={{$value}}',
                        },
                    },
                },
            },
        ],
        displayOptions,
    },
    {
        displayName: 'Search Filters',
        name: 'searchFilters',
        type: 'collection',
        placeholder: 'Add Search Filter',
        default: {},
        options: [
            {
                displayName: 'Content Filter',
                name: 'filter',
                type: 'options',
                default: '1',
                description: 'This parameter controls whether the \'Duplicate Content\' and \'Host Crowding\' filters are enabled. Set the value to 1 to enable these filters, which is the default setting. To disable these filters, set the value to 0.',
                options: [
                    { name: 'Disable Filters', value: '0' },
                    { name: 'Enable Filters', value: '1' },
                ],
                routing: {
                    request: {
                        qs: {
                            filter: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Country Restriction',
                name: 'cr',
                type: 'string',
                default: '',
                description: 'The cr parameter restricts search results to documents originating in a particular country. Google determines the country of a document by the top-level domain (TLD) of the document\'s URL or by Web server\'s IP address geographic location. Check the full list of supported Google cr countries.',
                routing: {
                    request: {
                        qs: {
                            cr: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Encoded Filters',
                name: 'tbs',
                type: 'string',
                default: '',
                description: 'This parameter restricts results to URLs based on encoded values. Parameter is normally constructed using size, color, image_type, time_period, usage_rights values. For instance, isz:l would return only results that has large image size.',
                routing: {
                    request: {
                        qs: {
                            tbs: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Language Restriction',
                name: 'lr',
                type: 'string',
                default: '',
                description: 'The lr parameter restricts search results to documents written in a particular language or a set of languages. The accepted format for this parameter is lang_{2-letter country code}. For instance, to filter documents written in Japanese, the value should be set to lang_jp. To incorporate multiple languages, a value like lang_it|lang_de restricts the search to documents written in either Italian or German. Google identifies the document language based on the top-level domain (TLD) of the document\'s URL, any language meta tags present, or the language utilized within the document\'s body text. Check the full list of supported Google lr languages.',
                routing: {
                    request: {
                        qs: {
                            lr: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'No Auto-Correction',
                name: 'nfpr',
                type: 'options',
                default: '0',
                description: 'This parameter controls whether results from queries that have been auto-corrected for spelling errors are included. To exclude these auto-corrected results, set the value to 1. By default, the value is 0, meaning auto-corrected results are included.',
                options: [
                    { name: 'Include Auto-Corrected Results', value: '0' },
                    { name: 'Exclude Auto-Corrected Results', value: '1' },
                ],
                routing: {
                    request: {
                        qs: {
                            nfpr: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'SafeSearch',
                name: 'safe',
                type: 'options',
                default: 'off',
                description: 'This parameter toggles the SafeSearch feature for the results. SafeSearch operates by filtering out adult content from your search results. Google\'s filters use proprietary technology to check keywords, phrases and URLs. While no filters are 100 percent accurate, SafeSearch will remove the overwhelming majority of adult content from your search results. Set the value to active to enable SafeSearch. To disable it, set the value to off. By default, SafeSearch is disabled.',
                options: [
                    { name: 'Active', value: 'active' },
                    { name: 'Off', value: 'off' },
                ],
                routing: {
                    request: {
                        qs: {
                            safe: '={{$value}}',
                        },
                    },
                },
            },
        ],
        displayOptions,
    },
    {
        displayName: 'Image Options',
        name: 'imageOptions',
        type: 'collection',
        placeholder: 'Add Image Option',
        default: {},
        options: [
            {
                displayName: 'Aspect Ratio',
                name: 'aspect_ratio',
                type: 'options',
                default: 'panoramic',
                description: 'This parameter filters images based on aspect ratio. Supported values are: square - width equals height. tall - height greater than width. wide - width greater than height. panoramic - width is at least twice the height.',
                options: [
                    { name: 'Panoramic', value: 'panoramic' },
                    { name: 'Square', value: 'square' },
                    { name: 'Tall', value: 'tall' },
                    { name: 'Wide', value: 'wide' },
                ],
                routing: {
                    request: {
                        qs: {
                            aspect_ratio: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Color Filter',
                name: 'color',
                type: 'options',
                default: 'black',
                description: 'This parameter controls the color of your search results. These options are available: black_and_white, color, transparent, red, orange, yellow, green, teal, blue, purple, pink, white, gray, black, brown.',
                options: [
                    { name: 'Black', value: 'black' },
                    { name: 'Black and White', value: 'black_and_white' },
                    { name: 'Blue', value: 'blue' },
                    { name: 'Brown', value: 'brown' },
                    { name: 'Color', value: 'color' },
                    { name: 'Gray', value: 'gray' },
                    { name: 'Green', value: 'green' },
                    { name: 'Orange', value: 'orange' },
                    { name: 'Pink', value: 'pink' },
                    { name: 'Purple', value: 'purple' },
                    { name: 'Red', value: 'red' },
                    { name: 'Teal', value: 'teal' },
                    { name: 'Transparent', value: 'transparent' },
                    { name: 'White', value: 'white' },
                    { name: 'Yellow', value: 'yellow' },
                ],
                routing: {
                    request: {
                        qs: {
                            color: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Image Size',
                name: 'size',
                type: 'options',
                default: 'icon',
                description: 'This parameter controls the size of your search results. There are few options available: large, medium, icon, larger_than_400x300, larger_than_640x480, larger_than_800x600, larger_than_1024x768, larger_than_2mp, larger_than_4mp, larger_than_6mp, larger_than_8mp, larger_than_12mp, larger_than_15mp, larger_than_20mp, larger_than_40mp, larger_than_70mp.',
                options: [
                    { name: 'Icon', value: 'icon' },
                    { name: 'Large', value: 'large' },
                    { name: 'Larger Than 1024x768', value: 'larger_than_1024x768' },
                    { name: 'Larger Than 12MP', value: 'larger_than_12mp' },
                    { name: 'Larger Than 15MP', value: 'larger_than_15mp' },
                    { name: 'Larger Than 20MP', value: 'larger_than_20mp' },
                    { name: 'Larger Than 2MP', value: 'larger_than_2mp' },
                    { name: 'Larger Than 400x300', value: 'larger_than_400x300' },
                    { name: 'Larger Than 40MP', value: 'larger_than_40mp' },
                    { name: 'Larger Than 4MP', value: 'larger_than_4mp' },
                    { name: 'Larger Than 640x480', value: 'larger_than_640x480' },
                    { name: 'Larger Than 6MP', value: 'larger_than_6mp' },
                    { name: 'Larger Than 70MP', value: 'larger_than_70mp' },
                    { name: 'Larger Than 800x600', value: 'larger_than_800x600' },
                    { name: 'Larger Than 8MP', value: 'larger_than_8mp' },
                    { name: 'Medium', value: 'medium' },
                ],
                routing: {
                    request: {
                        qs: {
                            size: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Image Type',
                name: 'image_type',
                type: 'options',
                default: 'clipart',
                description: 'This parameter controls the type of your search results. There are only few options that are available: clipart, line_drawing, gif, face, photo.',
                options: [
                    { name: 'Clipart', value: 'clipart' },
                    { name: 'Face', value: 'face' },
                    { name: 'GIF', value: 'gif' },
                    { name: 'Line Drawing', value: 'line_drawing' },
                    { name: 'Photo', value: 'photo' },
                ],
                routing: {
                    request: {
                        qs: {
                            image_type: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Time Period',
                name: 'time_period',
                type: 'options',
                default: 'last_day',
                description: 'This parameter restricts results to URLs based on date. Supported values are: last_hour, last_day, last_week, last_month, last_year.',
                options: [
                    { name: 'Last Day', value: 'last_day' },
                    { name: 'Last Hour', value: 'last_hour' },
                    { name: 'Last Month', value: 'last_month' },
                    { name: 'Last Week', value: 'last_week' },
                    { name: 'Last Year', value: 'last_year' },
                ],
                routing: {
                    request: {
                        qs: {
                            time_period: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Usage Rights',
                name: 'usage_rights',
                type: 'options',
                default: 'commercial_or_other_licenses',
                description: 'This parameter controls the usage rights of your search results. Options that are available: creative_commons_licenses, commercial_or_other_licenses.',
                options: [
                    { name: 'Commercial or Other Licenses', value: 'commercial_or_other_licenses' },
                    { name: 'Creative Commons Licenses', value: 'creative_commons_licenses' },
                ],
                routing: {
                    request: {
                        qs: {
                            usage_rights: '={{$value}}',
                        },
                    },
                },
            },
        ],
        displayOptions,
    },
    {
        displayName: 'Technical Settings',
        name: 'technicalSettings',
        type: 'collection',
        placeholder: 'Add Technical Setting',
        default: {},
        options: [
            {
                displayName: 'Device Type',
                name: 'device',
                type: 'options',
                default: 'desktop',
                description: 'The default parameter desktop defines the search on a desktop device. The mobile parameter defines the search on a mobile device. The tablet parameter defines the search on a tablet device.',
                options: [
                    { name: 'Desktop', value: 'desktop' },
                    { name: 'Mobile', value: 'mobile' },
                    { name: 'Tablet', value: 'tablet' },
                ],
                routing: {
                    request: {
                        qs: {
                            device: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Google Domain',
                name: 'google_domain',
                type: 'string',
                default: 'google.com',
                description: 'The default parameter google.com defines the Google domain of the search. Check the full list of supported Google google_domain domains.',
                routing: {
                    request: {
                        qs: {
                            google_domain: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Page Number',
                name: 'page',
                type: 'string',
                default: '1',
                description: 'This parameter indicates which page of results to return. By default, it is set to 1.',
                routing: {
                    request: {
                        qs: {
                            page: '={{$value}}',
                        },
                    },
                },
            },
        ],
        displayOptions,
    },
];
exports.google_images = {
    resource,
    properties,
};
