// @searchapi/n8n-nodes-searchapi/dist/nodes/SearchApi/engines/google_maps.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.google_maps = void 0;
const displayOptions = {
    show: {
        resource: ['google_maps'],
    },
};
const resource = {
    name: 'Google Maps',
    value: 'google_maps'
};
const properties = [
    {
        displayName: 'Query',
        name: 'q',
        type: 'string',
        required: true,
        default: '',
        description: 'Parameter defines the query you want to search. You can use anything that you would use in a regular Google Maps search.',
        options: [],
        displayOptions,
        routing: {
            request: {
                qs: {
                    q: '={{$value}}',
                },
            },
        },
    },
    {
        displayName: 'Location Settings',
        name: 'locationSettings',
        type: 'collection',
        placeholder: 'Add Location Setting',
        default: {},
        options: [
            {
                displayName: 'GPS Coordinates',
                name: 'll',
                type: 'string',
                default: '',
                description: 'The parameter specifies the GPS coordinates for the location where the query (q) should be applied. It can be formatted in two ways: @latitude,longitude,zoom (e.g. @40.7009973,-73.994778,12z) or @latitude,longitude,meters (e.g. @40.7009973,-73.994778,500m). The last value can end with z (zoom, from 3z to 21z) or m (meters for radius, from 62m to 18636559m). While the zoom parameter is optional, it is advisable to use it for greater accuracy.',
                options: [],
                routing: {
                    request: {
                        qs: {
                            ll: '={{$value}}',
                        },
                    },
                },
            },
        ],
        displayOptions,
    },
    {
        displayName: 'Search Configuration',
        name: 'searchConfiguration',
        type: 'collection',
        placeholder: 'Add Search Configuration',
        default: {},
        options: [
            {
                displayName: 'Google Domain',
                name: 'google_domain',
                type: 'options',
                default: 'google.com',
                description: 'The default parameter google.com defines the Google domain of the search. Check the full list of supported Google google_domain domains.',
                options: [
                    { name: 'Google.ad', value: 'google.ad' },
                    { name: 'Google.ae', value: 'google.ae' },
                    { name: 'Google.al', value: 'google.al' },
                    { name: 'Google.am', value: 'google.am' },
                    { name: 'Google.as', value: 'google.as' },
                    { name: 'Google.at', value: 'google.at' },
                    { name: 'Google.az', value: 'google.az' },
                    { name: 'Google.ba', value: 'google.ba' },
                    { name: 'Google.be', value: 'google.be' },
                    { name: 'Google.bf', value: 'google.bf' },
                    { name: 'Google.bg', value: 'google.bg' },
                    { name: 'Google.bi', value: 'google.bi' },
                    { name: 'Google.bj', value: 'google.bj' },
                    { name: 'Google.bs', value: 'google.bs' },
                    { name: 'Google.bt', value: 'google.bt' },
                    { name: 'Google.by', value: 'google.by' },
                    { name: 'Google.ca', value: 'google.ca' },
                    { name: 'Google.cd', value: 'google.cd' },
                    { name: 'Google.cf', value: 'google.cf' },
                    { name: 'Google.cg', value: 'google.cg' },
                    { name: 'Google.ch', value: 'google.ch' },
                    { name: 'Google.ci', value: 'google.ci' },
                    { name: 'Google.cl', value: 'google.cl' },
                    { name: 'Google.cm', value: 'google.cm' },
                    { name: 'Google.co.ao', value: 'google.co.ao' },
                    { name: 'Google.co.bw', value: 'google.co.bw' },
                    { name: 'Google.co.ck', value: 'google.co.ck' },
                    { name: 'Google.co.cr', value: 'google.co.cr' },
                    { name: 'Google.co.ID', value: 'google.co.id' },
                    { name: 'Google.co.il', value: 'google.co.il' },
                    { name: 'Google.co.in', value: 'google.co.in' },
                    { name: 'Google.co.jp', value: 'google.co.jp' },
                    { name: 'Google.co.ke', value: 'google.co.ke' },
                    { name: 'Google.co.kr', value: 'google.co.kr' },
                    { name: 'Google.co.ls', value: 'google.co.ls' },
                    { name: 'Google.co.ma', value: 'google.co.ma' },
                    { name: 'Google.co.mz', value: 'google.co.mz' },
                    { name: 'Google.co.nz', value: 'google.co.nz' },
                    { name: 'Google.co.th', value: 'google.co.th' },
                    { name: 'Google.co.tz', value: 'google.co.tz' },
                    { name: 'Google.co.ug', value: 'google.co.ug' },
                    { name: 'Google.co.uk', value: 'google.co.uk' },
                    { name: 'Google.co.uz', value: 'google.co.uz' },
                    { name: 'Google.co.ve', value: 'google.co.ve' },
                    { name: 'Google.co.vi', value: 'google.co.vi' },
                    { name: 'Google.co.za', value: 'google.co.za' },
                    { name: 'Google.co.zw', value: 'google.co.zw' },
                    { name: 'Google.com', value: 'google.com' },
                    { name: 'Google.com.af', value: 'google.com.af' },
                    { name: 'Google.com.ag', value: 'google.com.ag' },
                    { name: 'Google.com.ai', value: 'google.com.ai' },
                    { name: 'Google.com.ar', value: 'google.com.ar' },
                    { name: 'Google.com.au', value: 'google.com.au' },
                    { name: 'Google.com.bd', value: 'google.com.bd' },
                    { name: 'Google.com.bh', value: 'google.com.bh' },
                    { name: 'Google.com.bn', value: 'google.com.bn' },
                    { name: 'Google.com.bo', value: 'google.com.bo' },
                    { name: 'Google.com.br', value: 'google.com.br' },
                    { name: 'Google.com.bz', value: 'google.com.bz' },
                    { name: 'Google.com.co', value: 'google.com.co' },
                    { name: 'Google.com.cu', value: 'google.com.cu' },
                    { name: 'Google.com.cy', value: 'google.com.cy' },
                    { name: 'Google.com.do', value: 'google.com.do' },
                    { name: 'Google.com.ec', value: 'google.com.ec' },
                    { name: 'Google.com.eg', value: 'google.com.eg' },
                    { name: 'Google.com.et', value: 'google.com.et' },
                    { name: 'Google.com.fj', value: 'google.com.fj' },
                    { name: 'Google.com.gh', value: 'google.com.gh' },
                    { name: 'Google.com.gi', value: 'google.com.gi' },
                    { name: 'Google.com.gt', value: 'google.com.gt' },
                    { name: 'Google.com.hk', value: 'google.com.hk' },
                    { name: 'Google.com.jm', value: 'google.com.jm' },
                    { name: 'Google.com.kh', value: 'google.com.kh' },
                    { name: 'Google.com.kn', value: 'google.com.kn' },
                    { name: 'Google.com.kw', value: 'google.com.kw' },
                    { name: 'Google.com.lb', value: 'google.com.lb' },
                    { name: 'Google.com.lc', value: 'google.com.lc' },
                    { name: 'Google.com.ly', value: 'google.com.ly' },
                    { name: 'Google.com.mm', value: 'google.com.mm' },
                    { name: 'Google.com.mt', value: 'google.com.mt' },
                    { name: 'Google.com.mx', value: 'google.com.mx' },
                    { name: 'Google.com.my', value: 'google.com.my' },
                    { name: 'Google.com.na', value: 'google.com.na' },
                    { name: 'Google.com.ng', value: 'google.com.ng' },
                    { name: 'Google.com.ni', value: 'google.com.ni' },
                    { name: 'Google.com.np', value: 'google.com.np' },
                    { name: 'Google.com.om', value: 'google.com.om' },
                    { name: 'Google.com.pa', value: 'google.com.pa' },
                    { name: 'Google.com.pe', value: 'google.com.pe' },
                    { name: 'Google.com.pg', value: 'google.com.pg' },
                    { name: 'Google.com.ph', value: 'google.com.ph' },
                    { name: 'Google.com.pk', value: 'google.com.pk' },
                    { name: 'Google.com.pr', value: 'google.com.pr' },
                    { name: 'Google.com.py', value: 'google.com.py' },
                    { name: 'Google.com.qa', value: 'google.com.qa' },
                    { name: 'Google.com.sa', value: 'google.com.sa' },
                    { name: 'Google.com.sb', value: 'google.com.sb' },
                    { name: 'Google.com.sg', value: 'google.com.sg' },
                    { name: 'Google.com.sl', value: 'google.com.sl' },
                    { name: 'Google.com.tj', value: 'google.com.tj' },
                    { name: 'Google.com.tr', value: 'google.com.tr' },
                    { name: 'Google.com.ua', value: 'google.com.ua' },
                    { name: 'Google.com.uy', value: 'google.com.uy' },
                    { name: 'Google.com.vc', value: 'google.com.vc' },
                    { name: 'Google.com.vg', value: 'google.com.vg' },
                    { name: 'Google.cv', value: 'google.cv' },
                    { name: 'Google.cz', value: 'google.cz' },
                    { name: 'Google.de', value: 'google.de' },
                    { name: 'Google.dj', value: 'google.dj' },
                    { name: 'Google.dk', value: 'google.dk' },
                    { name: 'Google.dm', value: 'google.dm' },
                    { name: 'Google.dz', value: 'google.dz' },
                    { name: 'Google.ee', value: 'google.ee' },
                    { name: 'Google.eh', value: 'google.eh' },
                    { name: 'Google.es', value: 'google.es' },
                    { name: 'Google.fi', value: 'google.fi' },
                    { name: 'Google.fm', value: 'google.fm' },
                    { name: 'Google.fr', value: 'google.fr' },
                    { name: 'Google.ga', value: 'google.ga' },
                    { name: 'Google.ge', value: 'google.ge' },
                    { name: 'Google.gg', value: 'google.gg' },
                    { name: 'Google.gl', value: 'google.gl' },
                    { name: 'Google.gm', value: 'google.gm' },
                    { name: 'Google.gp', value: 'google.gp' },
                    { name: 'Google.gr', value: 'google.gr' },
                    { name: 'Google.gs', value: 'google.gs' },
                    { name: 'Google.gy', value: 'google.gy' },
                    { name: 'Google.hn', value: 'google.hn' },
                    { name: 'Google.hr', value: 'google.hr' },
                    { name: 'Google.ht', value: 'google.ht' },
                    { name: 'Google.hu', value: 'google.hu' },
                    { name: 'Google.ie', value: 'google.ie' },
                    { name: 'Google.iq', value: 'google.iq' },
                    { name: 'Google.is', value: 'google.is' },
                    { name: 'Google.it', value: 'google.it' },
                    { name: 'Google.jo', value: 'google.jo' },
                    { name: 'Google.kg', value: 'google.kg' },
                    { name: 'Google.ki', value: 'google.ki' },
                    { name: 'Google.kz', value: 'google.kz' },
                    { name: 'Google.la', value: 'google.la' },
                    { name: 'Google.li', value: 'google.li' },
                    { name: 'Google.lk', value: 'google.lk' },
                    { name: 'Google.lt', value: 'google.lt' },
                    { name: 'Google.lu', value: 'google.lu' },
                    { name: 'Google.lv', value: 'google.lv' },
                    { name: 'Google.md', value: 'google.md' },
                    { name: 'Google.mg', value: 'google.mg' },
                    { name: 'Google.mk', value: 'google.mk' },
                    { name: 'Google.ml', value: 'google.ml' },
                    { name: 'Google.mn', value: 'google.mn' },
                    { name: 'Google.ms', value: 'google.ms' },
                    { name: 'Google.mu', value: 'google.mu' },
                    { name: 'Google.mv', value: 'google.mv' },
                    { name: 'Google.mw', value: 'google.mw' },
                    { name: 'Google.ne', value: 'google.ne' },
                    { name: 'Google.nl', value: 'google.nl' },
                    { name: 'Google.no', value: 'google.no' },
                    { name: 'Google.nr', value: 'google.nr' },
                    { name: 'Google.nu', value: 'google.nu' },
                    { name: 'Google.pl', value: 'google.pl' },
                    { name: 'Google.pm', value: 'google.pm' },
                    { name: 'Google.pn', value: 'google.pn' },
                    { name: 'Google.ps', value: 'google.ps' },
                    { name: 'Google.pt', value: 'google.pt' },
                    { name: 'Google.pw', value: 'google.pw' },
                    { name: 'Google.re', value: 'google.re' },
                    { name: 'Google.ro', value: 'google.ro' },
                    { name: 'Google.rs', value: 'google.rs' },
                    { name: 'Google.ru', value: 'google.ru' },
                    { name: 'Google.rw', value: 'google.rw' },
                    { name: 'Google.sc', value: 'google.sc' },
                    { name: 'Google.sd', value: 'google.sd' },
                    { name: 'Google.se', value: 'google.se' },
                    { name: 'Google.sh', value: 'google.sh' },
                    { name: 'Google.si', value: 'google.si' },
                    { name: 'Google.sj', value: 'google.sj' },
                    { name: 'Google.sk', value: 'google.sk' },
                    { name: 'Google.sm', value: 'google.sm' },
                    { name: 'Google.sn', value: 'google.sn' },
                    { name: 'Google.so', value: 'google.so' },
                    { name: 'Google.sr', value: 'google.sr' },
                    { name: 'Google.st', value: 'google.st' },
                    { name: 'Google.sy', value: 'google.sy' },
                    { name: 'Google.sz', value: 'google.sz' },
                    { name: 'Google.tc', value: 'google.tc' },
                    { name: 'Google.tg', value: 'google.tg' },
                    { name: 'Google.tk', value: 'google.tk' },
                    { name: 'Google.tl', value: 'google.tl' },
                    { name: 'Google.tm', value: 'google.tm' },
                    { name: 'Google.tn', value: 'google.tn' },
                    { name: 'Google.to', value: 'google.to' },
                    { name: 'Google.tt', value: 'google.tt' },
                    { name: 'Google.tv', value: 'google.tv' },
                    { name: 'Google.tw', value: 'google.tw' },
                    { name: 'Google.vn', value: 'google.vn' },
                    { name: 'Google.vu', value: 'google.vu' },
                    { name: 'Google.wf', value: 'google.wf' },
                    { name: 'Google.ws', value: 'google.ws' },
                    { name: 'Google.ye', value: 'google.ye' },
                    { name: 'Google.zm', value: 'google.zm' },
                ],
                routing: {
                    request: {
                        qs: {
                            google_domain: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Interface Language',
                name: 'hl',
                type: 'options',
                default: 'en',
                description: 'The default parameter en defines the interface language of the search. Check the full list of supported Google hl languages.',
                options: [
                    { name: 'Afrikaans', value: 'af' },
                    { name: 'Akan', value: 'ak' },
                    { name: 'Albanian', value: 'sq' },
                    { name: 'Amharic', value: 'am' },
                    { name: 'Arabic', value: 'ar' },
                    { name: 'Armenian', value: 'hy' },
                    { name: 'Azerbaijani', value: 'az' },
                    { name: 'Basque', value: 'eu' },
                    { name: 'Belarusian', value: 'be' },
                    { name: 'Bemba', value: 'bem' },
                    { name: 'Bengali', value: 'bn' },
                    { name: 'Bihari', value: 'bh' },
                    { name: 'Bork', value: 'xx-bork' },
                    { name: 'Bosnian', value: 'bs' },
                    { name: 'Breton', value: 'br' },
                    { name: 'Bulgarian', value: 'bg' },
                    { name: 'Cambodian', value: 'km' },
                    { name: 'Catalan', value: 'ca' },
                    { name: 'Cherokee', value: 'chr' },
                    { name: 'Chinese (Simplified)', value: 'zh-cn' },
                    { name: 'Chinese (Traditional)', value: 'zh-tw' },
                    { name: 'Chinyanja', value: 'ny' },
                    { name: 'Corsican', value: 'co' },
                    { name: 'Croatian', value: 'hr' },
                    { name: 'Czech', value: 'cs' },
                    { name: 'Danish', value: 'da' },
                    { name: 'Dutch', value: 'nl' },
                    { name: 'Elmer Fudd', value: 'xx-elmer' },
                    { name: 'English', value: 'en' },
                    { name: 'Esperanto', value: 'eo' },
                    { name: 'Estonian', value: 'et' },
                    { name: 'Ewe', value: 'ee' },
                    { name: 'Faroese', value: 'fo' },
                    { name: 'Filipino', value: 'tl' },
                    { name: 'Finnish', value: 'fi' },
                    { name: 'French', value: 'fr' },
                    { name: 'Frisian', value: 'fy' },
                    { name: 'Ga', value: 'gaa' },
                    { name: 'Galician', value: 'gl' },
                    { name: 'Georgian', value: 'ka' },
                    { name: 'German', value: 'de' },
                    { name: 'Greek', value: 'el' },
                    { name: 'Greenlandic', value: 'kl' },
                    { name: 'Guarani', value: 'gn' },
                    { name: 'Gujarati', value: 'gu' },
                    { name: 'Hacker', value: 'xx-hacker' },
                    { name: 'Haitian Creole', value: 'ht' },
                    { name: 'Hausa', value: 'ha' },
                    { name: 'Hawaiian', value: 'haw' },
                    { name: 'Hebrew', value: 'iw' },
                    { name: 'Hindi', value: 'hi' },
                    { name: 'Hungarian', value: 'hu' },
                    { name: 'Icelandic', value: 'is' },
                    { name: 'Igbo', value: 'ig' },
                    { name: 'Indonesian', value: 'id' },
                    { name: 'Interlingua', value: 'ia' },
                    { name: 'Irish', value: 'ga' },
                    { name: 'Italian', value: 'it' },
                    { name: 'Japanese', value: 'ja' },
                    { name: 'Javanese', value: 'jw' },
                    { name: 'Kannada', value: 'kn' },
                    { name: 'Kazakh', value: 'kk' },
                    { name: 'Kinyarwanda', value: 'rw' },
                    { name: 'Kirundi', value: 'rn' },
                    { name: 'Klingon', value: 'xx-klingon' },
                    { name: 'Kongo', value: 'kg' },
                    { name: 'Korean', value: 'ko' },
                    { name: 'Krio', value: 'kri' },
                    { name: 'Kurdish', value: 'ku' },
                    { name: 'Kurdish (Sorani)', value: 'ckb' },
                    { name: 'Kyrgyz', value: 'ky' },
                    { name: 'Lao', value: 'lo' },
                    { name: 'Latin', value: 'la' },
                    { name: 'Latvian', value: 'lv' },
                    { name: 'Lingala', value: 'ln' },
                    { name: 'Lithuanian', value: 'lt' },
                    { name: 'Lozi', value: 'loz' },
                    { name: 'Luganda', value: 'lg' },
                    { name: 'Luo', value: 'ach' },
                    { name: 'Macedonian', value: 'mk' },
                    { name: 'Malagasy', value: 'mg' },
                    { name: 'Malay', value: 'ms' },
                    { name: 'Malayalam', value: 'ml' },
                    { name: 'Maltese', value: 'mt' },
                    { name: 'Māori', value: 'mi' },
                    { name: 'Marathi', value: 'mr' },
                    { name: 'Mauritian Creole', value: 'mfe' },
                    { name: 'Moldavian', value: 'mo' },
                    { name: 'Mongolian', value: 'mn' },
                    { name: 'Montenegrin', value: 'sr-me' },
                    { name: 'Myanmar (Burmese)', value: 'my' },
                    { name: 'Nepali', value: 'ne' },
                    { name: 'Nigerian Pidgin', value: 'pcm' },
                    { name: 'Northern Sotho', value: 'nso' },
                    { name: 'Norwegian', value: 'no' },
                    { name: 'Norwegian (Nynorsk)', value: 'nn' },
                    { name: 'Occitan', value: 'oc' },
                    { name: 'Oriya', value: 'or' },
                    { name: 'Oromo', value: 'om' },
                    { name: 'Pashto', value: 'ps' },
                    { name: 'Persian', value: 'fa' },
                    { name: 'Pirate', value: 'xx-pirate' },
                    { name: 'Polish', value: 'pl' },
                    { name: 'Portuguese', value: 'pt' },
                    { name: 'Portuguese (Brazil)', value: 'pt-br' },
                    { name: 'Portuguese (Portugal)', value: 'pt-pt' },
                    { name: 'Punjabi', value: 'pa' },
                    { name: 'Quechua', value: 'qu' },
                    { name: 'Romanian', value: 'ro' },
                    { name: 'Romansh', value: 'rm' },
                    { name: 'Runyankole', value: 'nyn' },
                    { name: 'Russian', value: 'ru' },
                    { name: 'Scottish Gaelic', value: 'gd' },
                    { name: 'Serbian', value: 'sr' },
                    { name: 'Serbo-Croatian', value: 'sh' },
                    { name: 'Sesotho', value: 'st' },
                    { name: 'Setswana', value: 'tn' },
                    { name: 'Seychellois Creole', value: 'crs' },
                    { name: 'Shona', value: 'sn' },
                    { name: 'Sindhi', value: 'sd' },
                    { name: 'Sinhalese', value: 'si' },
                    { name: 'Slovak', value: 'sk' },
                    { name: 'Slovenian', value: 'sl' },
                    { name: 'Somali', value: 'so' },
                    { name: 'Spanish', value: 'es' },
                    { name: 'Spanish (Latin America)', value: 'es-419' },
                    { name: 'Sundanese', value: 'su' },
                    { name: 'Swahili', value: 'sw' },
                    { name: 'Swedish', value: 'sv' },
                    { name: 'Tajik', value: 'tg' },
                    { name: 'Tamil', value: 'ta' },
                    { name: 'Tatar', value: 'tt' },
                    { name: 'Telugu', value: 'te' },
                    { name: 'Thai', value: 'th' },
                    { name: 'Tigrinya', value: 'ti' },
                    { name: 'Tongan', value: 'to' },
                    { name: 'Tshiluba', value: 'lua' },
                    { name: 'Tumbuka', value: 'tum' },
                    { name: 'Turkish', value: 'tr' },
                    { name: 'Turkmen', value: 'tk' },
                    { name: 'Twi', value: 'tw' },
                    { name: 'Ukrainian', value: 'uk' },
                    { name: 'Urdu', value: 'ur' },
                    { name: 'Uyghur', value: 'ug' },
                    { name: 'Uzbek', value: 'uz' },
                    { name: 'Vietnamese', value: 'vi' },
                    { name: 'Wolof', value: 'wo' },
                    { name: 'Xhosa', value: 'xh' },
                    { name: 'Yiddish', value: 'yi' },
                    { name: 'Yoruba', value: 'yo' },
                    { name: 'Zulu', value: 'zu' },
                ],
                routing: {
                    request: {
                        qs: {
                            hl: '={{$value}}',
                        },
                    },
                },
            },
        ],
        displayOptions,
    },
    {
        displayName: 'Pagination',
        name: 'pagination',
        type: 'collection',
        placeholder: 'Add Pagination Setting',
        default: {},
        options: [
            {
                displayName: 'Page Number',
                name: 'page',
                type: 'string',
                default: '1',
                description: 'This parameter indicates which page of results to return. By default, it is set to 1.',
                options: [],
                routing: {
                    request: {
                        qs: {
                            page: '={{$value}}',
                        },
                    },
                },
            },
        ],
        displayOptions,
    },
];
exports.google_maps = {
    resource,
    properties,
};
