// @searchapi/n8n-nodes-searchapi/dist/nodes/SearchApi/engines/google_shopping.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.google_shopping = void 0;
const displayOptions = {
    show: {
        resource: ['google_shopping'],
    },
};
const resource = {
    name: 'Google Shopping',
    value: 'google_shopping'
};
const properties = [
    {
        displayName: 'Query',
        name: 'q',
        type: 'string',
        default: '',
        required: true,
        description: 'Specifies the search query, similar to performing a regular Google search. Filters embedded in the q parameter act as hints rather than strict filters. If there are too few matching items, results outside the filters may appear. Examples of usage: Search by price: tshirt under $30, skinny fit kids jeans between $15 and $30, charizard over $300. Search by size: mens size 8.5 nike air. Search by color: Playstation white. Search nearby stores: Apparel nearby. Search by condition: iphone used, macbook new. Search by brand: prada dress, gucci shoes. Search discounted products: laptop on sale, headphones on sale.',
        options: [],
        displayOptions,
        routing: {
            request: {
                qs: {
                    q: '={{$value}}',
                },
            },
        },
    },
    {
        displayName: 'Location Settings',
        name: 'locationSettings',
        type: 'collection',
        placeholder: 'Add Location Setting',
        default: {},
        options: [
            {
                displayName: 'Geographic Location',
                name: 'location',
                type: 'string',
                default: '',
                description: 'Defines from where you want the search to originate. If several locations match the requested location, the most popular one is selected. Head to the Locations API for more precise control.',
                routing: {
                    request: {
                        qs: {
                            location: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Encoded Location',
                name: 'uule',
                type: 'string',
                default: '',
                description: 'Defines the Google encoded location for the search. SearchApi automatically generates the uule parameter when the location parameter is used, but it can be overwritten directly. uule and location cannot be used together.',
                routing: {
                    request: {
                        qs: {
                            uule: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Google Domain',
                name: 'google_domain',
                type: 'string',
                default: 'google.com',
                description: 'Defines the Google domain for the search. The default is google.com. Check the full list of supported Google google_domain domains.',
                routing: {
                    request: {
                        qs: {
                            google_domain: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Country',
                name: 'gl',
                type: 'options',
                default: 'us',
                description: 'The default parameter us defines the country of the search. Check the full list of supported Google gl countries.',
                options: [
                    { name: 'Ad', value: 'ad' },
                    { name: 'Ae', value: 'ae' },
                    { name: 'Af', value: 'af' },
                    { name: 'Ag', value: 'ag' },
                    { name: 'Ai', value: 'ai' },
                    { name: 'Al', value: 'al' },
                    { name: 'Am', value: 'am' },
                    { name: 'Ao', value: 'ao' },
                    { name: 'Aq', value: 'aq' },
                    { name: 'Ar', value: 'ar' },
                    { name: 'As', value: 'as' },
                    { name: 'At', value: 'at' },
                    { name: 'Au', value: 'au' },
                    { name: 'Aw', value: 'aw' },
                    { name: 'Az', value: 'az' },
                    { name: 'Ba', value: 'ba' },
                    { name: 'Bb', value: 'bb' },
                    { name: 'Bd', value: 'bd' },
                    { name: 'Be', value: 'be' },
                    { name: 'Bf', value: 'bf' },
                    { name: 'Bg', value: 'bg' },
                    { name: 'Bh', value: 'bh' },
                    { name: 'Bi', value: 'bi' },
                    { name: 'Bj', value: 'bj' },
                    { name: 'Bm', value: 'bm' },
                    { name: 'Bn', value: 'bn' },
                    { name: 'Bo', value: 'bo' },
                    { name: 'Br', value: 'br' },
                    { name: 'Bs', value: 'bs' },
                    { name: 'Bt', value: 'bt' },
                    { name: 'Bv', value: 'bv' },
                    { name: 'Bw', value: 'bw' },
                    { name: 'By', value: 'by' },
                    { name: 'Bz', value: 'bz' },
                    { name: 'Ca', value: 'ca' },
                    { name: 'Cc', value: 'cc' },
                    { name: 'Cd', value: 'cd' },
                    { name: 'Cf', value: 'cf' },
                    { name: 'Cg', value: 'cg' },
                    { name: 'Ch', value: 'ch' },
                    { name: 'Ci', value: 'ci' },
                    { name: 'Ck', value: 'ck' },
                    { name: 'Cl', value: 'cl' },
                    { name: 'Cm', value: 'cm' },
                    { name: 'Cn', value: 'cn' },
                    { name: 'Co', value: 'co' },
                    { name: 'Cr', value: 'cr' },
                    { name: 'Cu', value: 'cu' },
                    { name: 'Cv', value: 'cv' },
                    { name: 'Cx', value: 'cx' },
                    { name: 'Cy', value: 'cy' },
                    { name: 'Cz', value: 'cz' },
                    { name: 'De', value: 'de' },
                    { name: 'Dj', value: 'dj' },
                    { name: 'Dk', value: 'dk' },
                    { name: 'Dm', value: 'dm' },
                    { name: 'Do', value: 'do' },
                    { name: 'Dz', value: 'dz' },
                    { name: 'Ec', value: 'ec' },
                    { name: 'Ee', value: 'ee' },
                    { name: 'Eg', value: 'eg' },
                    { name: 'Eh', value: 'eh' },
                    { name: 'Er', value: 'er' },
                    { name: 'Es', value: 'es' },
                    { name: 'Et', value: 'et' },
                    { name: 'Fi', value: 'fi' },
                    { name: 'Fj', value: 'fj' },
                    { name: 'Fk', value: 'fk' },
                    { name: 'Fm', value: 'fm' },
                    { name: 'Fo', value: 'fo' },
                    { name: 'Fr', value: 'fr' },
                    { name: 'Ga', value: 'ga' },
                    { name: 'Gb', value: 'gb' },
                    { name: 'Gd', value: 'gd' },
                    { name: 'Ge', value: 'ge' },
                    { name: 'Gf', value: 'gf' },
                    { name: 'Gg', value: 'gg' },
                    { name: 'Gh', value: 'gh' },
                    { name: 'Gi', value: 'gi' },
                    { name: 'Gl', value: 'gl' },
                    { name: 'Gm', value: 'gm' },
                    { name: 'Gn', value: 'gn' },
                    { name: 'Gp', value: 'gp' },
                    { name: 'Gq', value: 'gq' },
                    { name: 'Gr', value: 'gr' },
                    { name: 'Gs', value: 'gs' },
                    { name: 'Gt', value: 'gt' },
                    { name: 'Gu', value: 'gu' },
                    { name: 'Gw', value: 'gw' },
                    { name: 'Gy', value: 'gy' },
                    { name: 'Hk', value: 'hk' },
                    { name: 'Hm', value: 'hm' },
                    { name: 'Hn', value: 'hn' },
                    { name: 'Hr', value: 'hr' },
                    { name: 'Ht', value: 'ht' },
                    { name: 'Hu', value: 'hu' },
                    { name: 'ID', value: 'id' },
                    { name: 'IE', value: 'ie' },
                    { name: 'IL', value: 'il' },
                    { name: 'IM', value: 'im' },
                    { name: 'IN', value: 'in' },
                    { name: 'IO', value: 'io' },
                    { name: 'Iq', value: 'iq' },
                    { name: 'Ir', value: 'ir' },
                    { name: 'Is', value: 'is' },
                    { name: 'It', value: 'it' },
                    { name: 'Je', value: 'je' },
                    { name: 'Jm', value: 'jm' },
                    { name: 'Jo', value: 'jo' },
                    { name: 'Jp', value: 'jp' },
                    { name: 'Ke', value: 'ke' },
                    { name: 'Kg', value: 'kg' },
                    { name: 'Kh', value: 'kh' },
                    { name: 'Ki', value: 'ki' },
                    { name: 'Km', value: 'km' },
                    { name: 'Kn', value: 'kn' },
                    { name: 'Kp', value: 'kp' },
                    { name: 'Kr', value: 'kr' },
                    { name: 'Kw', value: 'kw' },
                    { name: 'Ky', value: 'ky' },
                    { name: 'Kz', value: 'kz' },
                    { name: 'La', value: 'la' },
                    { name: 'Lb', value: 'lb' },
                    { name: 'Lc', value: 'lc' },
                    { name: 'Li', value: 'li' },
                    { name: 'Lk', value: 'lk' },
                    { name: 'Lr', value: 'lr' },
                    { name: 'Ls', value: 'ls' },
                    { name: 'Lt', value: 'lt' },
                    { name: 'Lu', value: 'lu' },
                    { name: 'Lv', value: 'lv' },
                    { name: 'Ly', value: 'ly' },
                    { name: 'Ma', value: 'ma' },
                    { name: 'Mc', value: 'mc' },
                    { name: 'Md', value: 'md' },
                    { name: 'Me', value: 'me' },
                    { name: 'Mg', value: 'mg' },
                    { name: 'Mh', value: 'mh' },
                    { name: 'Mk', value: 'mk' },
                    { name: 'Ml', value: 'ml' },
                    { name: 'Mm', value: 'mm' },
                    { name: 'Mn', value: 'mn' },
                    { name: 'Mo', value: 'mo' },
                    { name: 'Mp', value: 'mp' },
                    { name: 'Mq', value: 'mq' },
                    { name: 'Mr', value: 'mr' },
                    { name: 'Ms', value: 'ms' },
                    { name: 'Mt', value: 'mt' },
                    { name: 'Mu', value: 'mu' },
                    { name: 'Mv', value: 'mv' },
                    { name: 'Mw', value: 'mw' },
                    { name: 'Mx', value: 'mx' },
                    { name: 'My', value: 'my' },
                    { name: 'Mz', value: 'mz' },
                    { name: 'Na', value: 'na' },
                    { name: 'Nc', value: 'nc' },
                    { name: 'Ne', value: 'ne' },
                    { name: 'Nf', value: 'nf' },
                    { name: 'Ng', value: 'ng' },
                    { name: 'Ni', value: 'ni' },
                    { name: 'Nl', value: 'nl' },
                    { name: 'No', value: 'no' },
                    { name: 'Np', value: 'np' },
                    { name: 'Nr', value: 'nr' },
                    { name: 'Nu', value: 'nu' },
                    { name: 'Nz', value: 'nz' },
                    { name: 'Om', value: 'om' },
                    { name: 'Pa', value: 'pa' },
                    { name: 'Pe', value: 'pe' },
                    { name: 'Pf', value: 'pf' },
                    { name: 'Pg', value: 'pg' },
                    { name: 'Ph', value: 'ph' },
                    { name: 'Pk', value: 'pk' },
                    { name: 'Pl', value: 'pl' },
                    { name: 'Pm', value: 'pm' },
                    { name: 'Pn', value: 'pn' },
                    { name: 'Pr', value: 'pr' },
                    { name: 'Ps', value: 'ps' },
                    { name: 'Pt', value: 'pt' },
                    { name: 'Pw', value: 'pw' },
                    { name: 'Py', value: 'py' },
                    { name: 'Qa', value: 'qa' },
                    { name: 'Re', value: 're' },
                    { name: 'Ro', value: 'ro' },
                    { name: 'Rs', value: 'rs' },
                    { name: 'Ru', value: 'ru' },
                    { name: 'Rw', value: 'rw' },
                    { name: 'Sa', value: 'sa' },
                    { name: 'Sb', value: 'sb' },
                    { name: 'Sc', value: 'sc' },
                    { name: 'Sd', value: 'sd' },
                    { name: 'Se', value: 'se' },
                    { name: 'Sg', value: 'sg' },
                    { name: 'Sh', value: 'sh' },
                    { name: 'Si', value: 'si' },
                    { name: 'Sj', value: 'sj' },
                    { name: 'Sk', value: 'sk' },
                    { name: 'Sl', value: 'sl' },
                    { name: 'Sm', value: 'sm' },
                    { name: 'Sn', value: 'sn' },
                    { name: 'So', value: 'so' },
                    { name: 'Sr', value: 'sr' },
                    { name: 'St', value: 'st' },
                    { name: 'Sv', value: 'sv' },
                    { name: 'Sy', value: 'sy' },
                    { name: 'Sz', value: 'sz' },
                    { name: 'Tc', value: 'tc' },
                    { name: 'Td', value: 'td' },
                    { name: 'Tf', value: 'tf' },
                    { name: 'Tg', value: 'tg' },
                    { name: 'Th', value: 'th' },
                    { name: 'Tj', value: 'tj' },
                    { name: 'Tk', value: 'tk' },
                    { name: 'Tl', value: 'tl' },
                    { name: 'Tm', value: 'tm' },
                    { name: 'Tn', value: 'tn' },
                    { name: 'To', value: 'to' },
                    { name: 'Tr', value: 'tr' },
                    { name: 'Tt', value: 'tt' },
                    { name: 'Tv', value: 'tv' },
                    { name: 'Tw', value: 'tw' },
                    { name: 'Tz', value: 'tz' },
                    { name: 'Ua', value: 'ua' },
                    { name: 'Ug', value: 'ug' },
                    { name: 'Uk', value: 'uk' },
                    { name: 'Um', value: 'um' },
                    { name: 'Us', value: 'us' },
                    { name: 'Uy', value: 'uy' },
                    { name: 'Uz', value: 'uz' },
                    { name: 'Va', value: 'va' },
                    { name: 'Vc', value: 'vc' },
                    { name: 'Ve', value: 've' },
                    { name: 'Vg', value: 'vg' },
                    { name: 'Vi', value: 'vi' },
                    { name: 'Vn', value: 'vn' },
                    { name: 'Vu', value: 'vu' },
                    { name: 'Wf', value: 'wf' },
                    { name: 'Ws', value: 'ws' },
                    { name: 'Ye', value: 'ye' },
                    { name: 'Yt', value: 'yt' },
                    { name: 'Za', value: 'za' },
                    { name: 'Zm', value: 'zm' },
                    { name: 'Zw', value: 'zw' },
                ],
                routing: {
                    request: {
                        qs: {
                            gl: '={{$value}}',
                        },
                    },
                },
            },
        ],
        displayOptions,
    },
    {
        displayName: 'Language Settings',
        name: 'languageSettings',
        type: 'collection',
        placeholder: 'Add Language Setting',
        default: {},
        options: [
            {
                displayName: 'Interface Language',
                name: 'hl',
                type: 'options',
                default: 'en',
                description: 'The default parameter en defines the interface language of the search. Check the full list of supported Google hl languages.',
                options: [
                    { name: 'Ach', value: 'ach' },
                    { name: 'Af', value: 'af' },
                    { name: 'Ak', value: 'ak' },
                    { name: 'Am', value: 'am' },
                    { name: 'Ar', value: 'ar' },
                    { name: 'Az', value: 'az' },
                    { name: 'Be', value: 'be' },
                    { name: 'Bem', value: 'bem' },
                    { name: 'Bg', value: 'bg' },
                    { name: 'Bh', value: 'bh' },
                    { name: 'Bn', value: 'bn' },
                    { name: 'Br', value: 'br' },
                    { name: 'Bs', value: 'bs' },
                    { name: 'Ca', value: 'ca' },
                    { name: 'Chr', value: 'chr' },
                    { name: 'Ckb', value: 'ckb' },
                    { name: 'Co', value: 'co' },
                    { name: 'Crs', value: 'crs' },
                    { name: 'Cs', value: 'cs' },
                    { name: 'Cy', value: 'cy' },
                    { name: 'Da', value: 'da' },
                    { name: 'De', value: 'de' },
                    { name: 'Ee', value: 'ee' },
                    { name: 'El', value: 'el' },
                    { name: 'En', value: 'en' },
                    { name: 'Eo', value: 'eo' },
                    { name: 'Es', value: 'es' },
                    { name: 'Es-419', value: 'es-419' },
                    { name: 'Et', value: 'et' },
                    { name: 'Eu', value: 'eu' },
                    { name: 'Fa', value: 'fa' },
                    { name: 'Fi', value: 'fi' },
                    { name: 'Fo', value: 'fo' },
                    { name: 'Fr', value: 'fr' },
                    { name: 'Fy', value: 'fy' },
                    { name: 'Ga', value: 'ga' },
                    { name: 'Gaa', value: 'gaa' },
                    { name: 'Gd', value: 'gd' },
                    { name: 'Gl', value: 'gl' },
                    { name: 'Gn', value: 'gn' },
                    { name: 'Gu', value: 'gu' },
                    { name: 'Ha', value: 'ha' },
                    { name: 'Haw', value: 'haw' },
                    { name: 'Hi', value: 'hi' },
                    { name: 'Hr', value: 'hr' },
                    { name: 'Ht', value: 'ht' },
                    { name: 'Hu', value: 'hu' },
                    { name: 'Hy', value: 'hy' },
                    { name: 'IA', value: 'ia' },
                    { name: 'ID', value: 'id' },
                    { name: 'IG', value: 'ig' },
                    { name: 'IS', value: 'is' },
                    { name: 'IT', value: 'it' },
                    { name: 'IW', value: 'iw' },
                    { name: 'Ja', value: 'ja' },
                    { name: 'Jw', value: 'jw' },
                    { name: 'Ka', value: 'ka' },
                    { name: 'Kg', value: 'kg' },
                    { name: 'Kk', value: 'kk' },
                    { name: 'Kl', value: 'kl' },
                    { name: 'Km', value: 'km' },
                    { name: 'Kn', value: 'kn' },
                    { name: 'Ko', value: 'ko' },
                    { name: 'Kri', value: 'kri' },
                    { name: 'Ku', value: 'ku' },
                    { name: 'Ky', value: 'ky' },
                    { name: 'La', value: 'la' },
                    { name: 'Lg', value: 'lg' },
                    { name: 'Ln', value: 'ln' },
                    { name: 'Lo', value: 'lo' },
                    { name: 'Loz', value: 'loz' },
                    { name: 'Lt', value: 'lt' },
                    { name: 'Lua', value: 'lua' },
                    { name: 'Lv', value: 'lv' },
                    { name: 'Mfe', value: 'mfe' },
                    { name: 'Mg', value: 'mg' },
                    { name: 'Mi', value: 'mi' },
                    { name: 'Mk', value: 'mk' },
                    { name: 'Ml', value: 'ml' },
                    { name: 'Mn', value: 'mn' },
                    { name: 'Mo', value: 'mo' },
                    { name: 'Mp', value: 'mp' },
                    { name: 'Mq', value: 'mq' },
                    { name: 'Mr', value: 'mr' },
                    { name: 'Ms', value: 'ms' },
                    { name: 'Mt', value: 'mt' },
                    { name: 'Mv', value: 'mv' },
                    { name: 'Mw', value: 'mw' },
                    { name: 'Mx', value: 'mx' },
                    { name: 'My', value: 'my' },
                    { name: 'Mz', value: 'mz' },
                    { name: 'Na', value: 'na' },
                    { name: 'Nc', value: 'nc' },
                    { name: 'Ne', value: 'ne' },
                    { name: 'Nf', value: 'nf' },
                    { name: 'Ng', value: 'ng' },
                    { name: 'Ni', value: 'ni' },
                    { name: 'Nl', value: 'nl' },
                    { name: 'No', value: 'no' },
                    { name: 'Np', value: 'np' },
                    { name: 'Nr', value: 'nr' },
                    { name: 'Nu', value: 'nu' },
                    { name: 'Nz', value: 'nz' },
                    { name: 'Om', value: 'om' },
                    { name: 'Pa', value: 'pa' },
                    { name: 'Pe', value: 'pe' },
                    { name: 'Pf', value: 'pf' },
                    { name: 'Pg', value: 'pg' },
                    { name: 'Ph', value: 'ph' },
                    { name: 'Pk', value: 'pk' },
                    { name: 'Pl', value: 'pl' },
                    { name: 'Pm', value: 'pm' },
                    { name: 'Pn', value: 'pn' },
                    { name: 'Pr', value: 'pr' },
                    { name: 'Ps', value: 'ps' },
                    { name: 'Pt', value: 'pt' },
                    { name: 'PT-BR', value: 'pt-br' },
                    { name: 'PT-PT', value: 'pt-pt' },
                    { name: 'Qu', value: 'qu' },
                    { name: 'Rm', value: 'rm' },
                    { name: 'Rn', value: 'rn' },
                    { name: 'Ro', value: 'ro' },
                    { name: 'Ru', value: 'ru' },
                    { name: 'Rw', value: 'rw' },
                    { name: 'Sd', value: 'sd' },
                    { name: 'Sh', value: 'sh' },
                    { name: 'Si', value: 'si' },
                    { name: 'Sk', value: 'sk' },
                    { name: 'Sl', value: 'sl' },
                    { name: 'Sn', value: 'sn' },
                    { name: 'So', value: 'so' },
                    { name: 'Sq', value: 'sq' },
                    { name: 'Sr', value: 'sr' },
                    { name: 'SR-ME', value: 'sr-me' },
                    { name: 'St', value: 'st' },
                    { name: 'Su', value: 'su' },
                    { name: 'Sv', value: 'sv' },
                    { name: 'Sw', value: 'sw' },
                    { name: 'Ta', value: 'ta' },
                    { name: 'Te', value: 'te' },
                    { name: 'Tg', value: 'tg' },
                    { name: 'Th', value: 'th' },
                    { name: 'Ti', value: 'ti' },
                    { name: 'Tk', value: 'tk' },
                    { name: 'Tl', value: 'tl' },
                    { name: 'Tn', value: 'tn' },
                    { name: 'To', value: 'to' },
                    { name: 'Tr', value: 'tr' },
                    { name: 'Tt', value: 'tt' },
                    { name: 'Tum', value: 'tum' },
                    { name: 'Tw', value: 'tw' },
                    { name: 'UG', value: 'ug' },
                    { name: 'UK', value: 'uk' },
                    { name: 'UR', value: 'ur' },
                    { name: 'US', value: 'us' },
                    { name: 'UZ', value: 'uz' },
                    { name: 'VI', value: 'vi' },
                    { name: 'VU', value: 'vu' },
                    { name: 'WO', value: 'wo' },
                    { name: 'XH', value: 'xh' },
                    { name: 'XX-Bork', value: 'xx-bork' },
                    { name: 'XX-Elmer', value: 'xx-elmer' },
                    { name: 'XX-Hacker', value: 'xx-hacker' },
                    { name: 'XX-Klingon', value: 'xx-klingon' },
                    { name: 'XX-Pirate', value: 'xx-pirate' },
                    { name: 'Yi', value: 'yi' },
                    { name: 'Yo', value: 'yo' },
                    { name: 'ZH-CN', value: 'zh-cn' },
                    { name: 'ZH-TW', value: 'zh-tw' },
                    { name: 'Zu', value: 'zu' },
                ],
                routing: {
                    request: {
                        qs: {
                            hl: '={{$value}}',
                        },
                    },
                },
            },
        ],
        displayOptions,
    },
    {
        displayName: 'Price Filters',
        name: 'priceFilters',
        type: 'collection',
        placeholder: 'Add Price Filter',
        default: {},
        options: [
            {
                displayName: 'Minimum Price',
                name: 'price_min',
                type: 'string',
                default: '',
                description: 'Specifies the minimum price of the products returned. This parameter must include the currency. For example, 2.50 filters results to products priced at $2.50 or higher. Must be used as a strict filter before applying shoprs.',
                routing: {
                    request: {
                        qs: {
                            price_min: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Maximum Price',
                name: 'price_max',
                type: 'string',
                default: '',
                description: 'Specifies the maximum price of the products returned. This parameter must include the currency. For example, 100 filters results to products priced at $100 or less. Must be used as a strict filter before applying shoprs.',
                routing: {
                    request: {
                        qs: {
                            price_max: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Product Condition',
                name: 'condition',
                type: 'options',
                default: 'new',
                description: 'Filters product condition. Options include: new for new products, and used for used items.',
                options: [
                    { name: 'New', value: 'new' },
                    { name: 'Used', value: 'used' },
                ],
                routing: {
                    request: {
                        qs: {
                            condition: '={{$value}}',
                        },
                    },
                },
            },
        ],
        displayOptions,
    },
    {
        displayName: 'Advanced Options',
        name: 'advancedOptions',
        type: 'collection',
        placeholder: 'Add Advanced Option',
        default: {},
        options: [
            {
                displayName: 'Encoded Filters',
                name: 'shoprs',
                type: 'string',
                default: '',
                description: 'Applies strict filtering rules based on encoded values. Example: CAEYFyoDcHM1MhwIFxISUHJpY2U6IGxvdyB0byBoaWdoKgQQARgBYAKIAQE sorts products by price (low to high). Filters provided in the JSON response can be used in subsequent requests to apply additional filters iteratively.',
                routing: {
                    request: {
                        qs: {
                            shoprs: '={{$value}}',
                        },
                    },
                },
            },
            {
                displayName: 'Page Number',
                name: 'page',
                type: 'string',
                default: '1',
                description: 'Indicates which page of results to return. The default is 1.',
                routing: {
                    request: {
                        qs: {
                            page: '={{$value}}',
                        },
                    },
                },
            },
        ],
        displayOptions,
    },
];
exports.google_shopping = {
    resource,
    properties,
};
